--- cups-1.2rc1/man/Makefile.wiget	2006-04-05 12:06:41.000000000 +0200
+++ cups-1.2rc1/man/Makefile	2006-04-05 12:07:50.000000000 +0200
@@ -123,9 +123,9 @@
 		$(INSTALL_MAN) $$file $(AMANDIR)/man$(MAN8DIR); \
 	done
 	$(RM) $(AMANDIR)/man$(MAN8DIR)/reject.$(MAN8EXT)
-	$(LN) accept.$(MAN8EXT) $(AMANDIR)/man$(MAN8DIR)/reject.$(MAN8EXT)
+	echo ".so accept.$(MAN8EXT)" $(AMANDIR)/man$(MAN8DIR)/reject.$(MAN8EXT)
 	$(RM) $(AMANDIR)/man$(MAN8DIR)/cupsdisable.$(MAN8EXT)
-	$(LN) cupsenable.$(MAN8EXT) $(AMANDIR)/man$(MAN8DIR)/cupsdisable.$(MAN8EXT)
+	echo ".so cupsenable.$(MAN8EXT)" $(AMANDIR)/man$(MAN8DIR)/cupsdisable.$(MAN8EXT)
 
 
 #
