diff -urN cups-1.1.21.org/config-scripts/cups-sharedlibs.m4 cups-1.1.21/config-scripts/cups-sharedlibs.m4
--- cups-1.1.21.org/config-scripts/cups-sharedlibs.m4	2004-09-24 01:17:18.586596250 +0200
+++ cups-1.1.21/config-scripts/cups-sharedlibs.m4	2004-09-24 01:23:26.775720365 +0200
@@ -137,9 +137,11 @@
                         ;;
                 Linux | GNU)
                         # Linux and HURD
+			if test $libdir != /usr/lib -a $libdir != /usr/lib32 -a $libdir != /usr/lib64 -a $libdir != /lib ; then
                 	DSOFLAGS="-Wl,-rpath,$libdir $DSOFLAGS"
                         LDFLAGS="$LDFLAGS -Wl,-rpath,$libdir"
                         EXPORT_LDFLAGS="-Wl,-rpath,$libdir"
+			fi
                         ;;
 	esac
 else
