diff -urNp -x '*.orig' cups-2.3.3.org/config-scripts/cups-compiler.m4 cups-2.3.3/config-scripts/cups-compiler.m4
--- cups-2.3.3.org/config-scripts/cups-compiler.m4	2020-04-27 20:04:29.000000000 +0200
+++ cups-2.3.3/config-scripts/cups-compiler.m4	2021-04-20 22:45:49.129825106 +0200
@@ -30,8 +30,6 @@
 dnl For debugging, keep symbols, otherwise strip them...
 AS_IF([test x$enable_debug = xyes -a "x$OPTIM" = x], [
     OPTIM="-g"
-], [
-    INSTALL_STRIP="-s"
 ])
 
 dnl Debug printfs can slow things down, so provide a separate option for that
