--- cups-1.2.1/cups/http-addrlist.c.orig	2006-05-24 16:55:15.000000000 +0100
+++ cups-1.2.1/cups/http-addrlist.c	2006-05-24 16:56:11.000000000 +0100
@@ -506,6 +506,9 @@
 	temp->addr.ipv6.sin6_addr.s6_addr32[3] = htonl(1);
 #  endif /* WIN32 */
 
+        if (!first)
+          first = temp;
+
         addr = temp;
       }
 
@@ -527,6 +530,9 @@
 	temp->addr.ipv4.sin_port        = htons(portnum);
 	temp->addr.ipv4.sin_addr.s_addr = htonl(0x7f000001);
 
+        if (!first)
+          first = temp;
+
         if (addr)
 	  addr->next = temp;
 	else
@@ -556,6 +562,9 @@
         temp->addr.ipv6.sin6_family = AF_INET6;
 	temp->addr.ipv6.sin6_port   = htons(portnum);
 
+        if (!first)
+          first = temp;
+
         addr = temp;
       }
 
@@ -576,6 +585,9 @@
         temp->addr.ipv4.sin_family = AF_INET;
 	temp->addr.ipv4.sin_port   = htons(portnum);
 
+        if (!first)
+          first = temp;
+
         if (addr)
 	  addr->next = temp;
 	else
