--- cups-1.4b2/Makedefs.in~	2008-12-19 16:19:46.000000000 +0100
+++ cups-1.4b2/Makedefs.in	2008-12-19 16:20:36.000000000 +0100
@@ -124,7 +124,7 @@
 BACKLIBS	=	@BACKLIBS@
 BANNERTOPS	=	@BANNERTOPS@
 CFLAGS		=	@CPPFLAGS@ @CFLAGS@
-COMMONLIBS	=	@LIBS@
+COMMONLIBS	=	@LIBS@ $(DNSSDLIBS)
 CUPSDLIBS	=	@CUPSDLIBS@
 CXXFLAGS	=	@CPPFLAGS@ @CXXFLAGS@
 CXXLIBS		=	@CXXLIBS@
--- cups-1.4b2/scheduler/dirsvc.c~	2008-10-08 06:26:38.000000000 +0200
+++ cups-1.4b2/scheduler/dirsvc.c	2008-12-19 16:22:36.000000000 +0100
@@ -1948,7 +1948,7 @@
 
     WebIFRef = DNSSDRef;
     if ((error = DNSServiceRegister(&WebIFRef,
-				    kDNSServiceFlagsShareConnection,
+				    0,
 				    0, webif, "_http._tcp", NULL,
 				    NULL, htons(DNSSDPort), 7,
 				    "\006path=/", dnssdRegisterCallback,
@@ -2622,7 +2622,7 @@
     do
     {
       p->ipp_ref = DNSSDRef;
-      if ((se = DNSServiceRegister(&p->ipp_ref, kDNSServiceFlagsShareConnection,
+      if ((se = DNSServiceRegister(&p->ipp_ref, 0,
                                    0, name, regtype, domain, NULL,
 				   htons(DNSSDPort), ipp_len, ipp_txt,
 				   dnssdRegisterCallback,
@@ -2693,7 +2693,7 @@
 
       p->printer_ref = DNSSDRef;
       if ((se = DNSServiceRegister(&p->printer_ref,
-                                   kDNSServiceFlagsShareConnection,
+                                   0,
 				   0, name, "_printer._tcp", domain, NULL,
 				   htons(515), printer_len, printer_txt,
 				   dnssdRegisterCallback,
