--- cups-1.4b2/man/Makefile~	2008-12-17 14:19:01.000000000 +0100
+++ cups-1.4b2/man/Makefile	2008-12-17 14:17:59.000000000 +0100
@@ -152,10 +152,10 @@
 	done
 	for file in accept cupsreject reject; do \
 		$(RM) $(AMANDIR)/man$(MAN8DIR)/$$file.$(MAN8EXT); \
-		$(LN) cupsaccept.$(MAN8EXT) $(AMANDIR)/man$(MAN8DIR)/$$file.$(MAN8EXT); \
+		echo ".so cupsaccept.$(MAN8EXT)" $(AMANDIR)/man$(MAN8DIR)/$$file.$(MAN8EXT); \
 	done
 	$(RM) $(AMANDIR)/man$(MAN8DIR)/cupsdisable.$(MAN8EXT)
-	$(LN) cupsenable.$(MAN8EXT) $(AMANDIR)/man$(MAN8DIR)/cupsdisable.$(MAN8EXT)
+	echo ".so cupsenable.$(MAN8EXT)" $(AMANDIR)/man$(MAN8DIR)/cupsdisable.$(MAN8EXT)
 
 
 #
