diff -Nru cups-1.1.10/Makefile cups-1.1.10.new/Makefile
--- cups-1.1.10/Makefile	Mon Aug  6 21:37:08 2001
+++ cups-1.1.10.new/Makefile	Sun Sep  2 19:49:29 2001
@@ -74,24 +74,6 @@
 	(cd ppd; $(MAKE) $(MFLAGS) install)
 	echo Installing in templates...
 	(cd templates; $(MAKE) $(MFLAGS) install)
-	echo Installing startup script...
-	if test "x$(INITDIR)" != "x"; then \
-		$(INSTALL_DIR) $(prefix)/$(INITDIR)/init.d; \
-		$(INSTALL_SCRIPT) cups.sh $(prefix)/$(INITDIR)/init.d/cups; \
-		$(INSTALL_DIR) $(prefix)/$(INITDIR)/rc0.d; \
-		$(INSTALL_SCRIPT) cups.sh  $(prefix)/$(INITDIR)/rc0.d/K00cups; \
-		$(INSTALL_DIR) $(prefix)/$(INITDIR)/rc2.d; \
-		$(INSTALL_SCRIPT) cups.sh $(prefix)/$(INITDIR)/rc2.d/S99cups; \
-		$(INSTALL_DIR) $(prefix)/$(INITDIR)/rc3.d; \
-		$(INSTALL_SCRIPT) cups.sh $(prefix)/$(INITDIR)/rc3.d/S99cups; \
-		$(INSTALL_DIR) $(prefix)/$(INITDIR)/rc5.d; \
-		$(INSTALL_SCRIPT) cups.sh $(prefix)/$(INITDIR)/rc5.d/S99cups; \
-	fi
-	if test "x$(INITDIR)" = "x" -a "x$(INITDDIR)" != "x"; then \
-		$(INSTALL_DIR) $(prefix)/$(INITDDIR); \
-		$(INSTALL_SCRIPT) cups.sh $(prefix)/$(INITDDIR)/cups; \
-	fi
-
 
 #
 # Run the test suite...
diff -Nru cups-1.1.10/backend/Makefile cups-1.1.10.new/backend/Makefile
--- cups-1.1.10/backend/Makefile	Tue Jul 24 21:56:41 2001
+++ cups-1.1.10.new/backend/Makefile	Sun Sep  2 19:50:56 2001
@@ -49,12 +49,11 @@
 #
 
 install:
-	$(INSTALL_DIR) $(SERVERBIN)/backend
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/backend
 	for file in $(BACKENDS); do \
-		$(INSTALL_BIN) $$file $(SERVERBIN)/backend; \
+		$(INSTALL_BIN) $$file $(DESTDIR)$(SERVERBIN)/backend; \
 	done
-	$(RM) $(SERVERBIN)/backend/http
-	$(LN) ipp $(SERVERBIN)/backend/http
+	$(LN) -sf ipp $(DESTDIR)$(SERVERBIN)/backend/http
 
 
 #
diff -Nru cups-1.1.10/berkeley/Makefile cups-1.1.10.new/berkeley/Makefile
--- cups-1.1.10/berkeley/Makefile	Thu Jul 19 22:17:51 2001
+++ cups-1.1.10.new/berkeley/Makefile	Sun Sep  2 19:51:27 2001
@@ -48,12 +48,12 @@
 #
 
 install:
-	$(INSTALL_DIR) $(BINDIR)
-	$(INSTALL_BIN) lpq $(BINDIR)
-	$(INSTALL_BIN) lpr $(BINDIR)
-	$(INSTALL_BIN) lprm $(BINDIR)
-	$(INSTALL_DIR) $(SBINDIR)
-	$(INSTALL_BIN) lpc $(SBINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lpq $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lpr $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lprm $(DESTDIR)$(BINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) lpc $(DESTDIR)$(SBINDIR)
 
 
 #
diff -Nru cups-1.1.10/cgi-bin/Makefile cups-1.1.10.new/cgi-bin/Makefile
--- cups-1.1.10/cgi-bin/Makefile	Thu Jul 19 22:17:51 2001
+++ cups-1.1.10.new/cgi-bin/Makefile	Sun Sep  2 19:51:48 2001
@@ -50,9 +50,9 @@
 #
 
 install:
-	$(INSTALL_DIR) $(SERVERBIN)/cgi-bin
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/cgi-bin
 	for file in $(CGIS); do \
-		$(INSTALL_BIN) $$file $(SERVERBIN)/cgi-bin; \
+		$(INSTALL_BIN) $$file $(DESTDIR)$(SERVERBIN)/cgi-bin; \
 	done
 
 
diff -Nru cups-1.1.10/conf/Makefile cups-1.1.10.new/conf/Makefile
--- cups-1.1.10/conf/Makefile	Wed Jun 27 23:53:08 2001
+++ cups-1.1.10.new/conf/Makefile	Sun Sep  2 19:40:12 2001
@@ -51,19 +51,12 @@
 #
 
 install:
-	$(INSTALL_DIR) $(SERVERROOT)
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERROOT)
 	for file in $(KEEP); do \
-		if test -r $(SERVERROOT)/$$file ; then \
-			$(INSTALL_DATA) $$file $(SERVERROOT)/$$file.N ; \
-		else \
-			$(INSTALL_DATA) $$file $(SERVERROOT) ; \
-		fi ; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(SERVERROOT) ; \
 	done
 	for file in $(REPLACE); do \
-		if test -r $(SERVERROOT)/$$file ; then \
-			$(MV) $(SERVERROOT)/$$file $(SERVERROOT)/$$file.O ; \
-		fi ; \
-		$(INSTALL_DATA) $$file $(SERVERROOT) ; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(SERVERROOT) ; \
 	done
 
 
diff -Nru cups-1.1.10/cups/Makefile cups-1.1.10.new/cups/Makefile
--- cups-1.1.10/cups/Makefile	Mon Aug  6 21:37:09 2001
+++ cups-1.1.10.new/cups/Makefile	Sun Sep  2 19:53:05 2001
@@ -68,21 +68,14 @@
 #
 
 install:	all
-	$(INSTALL_DIR) $(INCLUDEDIR)/cups
+	$(INSTALL_DIR) $(DESTDIR)$(INCLUDEDIR)/cups
 	for file in $(HEADERS); do \
-		$(INSTALL_DATA) $$file $(INCLUDEDIR)/cups; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(INCLUDEDIR)/cups; \
 	done
-	$(INSTALL_DIR) $(LIBDIR)
-	$(INSTALL_LIB) $(LIBCUPS) $(LIBDIR)
-	if test $(LIBCUPS) = "libcups.so.2" -o $(LIBCUPS) = "libcups.sl.2"; then \
-		$(INSTALL_LIB) libcups.a $(LIBDIR); \
-		$(RM) $(LIBDIR)/`basename $(LIBCUPS) .2`; \
-		$(LN) $(LIBCUPS) $(LIBDIR)/`basename $(LIBCUPS) .2`; \
-	fi
-	if test $(LIBCUPS) = "libcups_s.a"; then \
-		$(INSTALL_LIB) libcups.a $(LIBDIR); \
-	fi
-
+	$(INSTALL_DIR) $(DESTDIR)$(LIBDIR)
+	$(INSTALL_LIB) $(LIBCUPS) $(DESTDIR)$(LIBDIR)
+	$(LN) $(LIBCUPS) $(DESTDIR)$(LIBDIR)/`basename $(LIBCUPS) .2`
+	$(INSTALL_LIB) libcups.a $(DESTDIR)$(LIBDIR)
 
 #
 # libcups.so.2, libcups.sl.2
diff -Nru cups-1.1.10/data/Makefile cups-1.1.10.new/data/Makefile
--- cups-1.1.10/data/Makefile	Thu Jul 19 22:17:52 2001
+++ cups-1.1.10.new/data/Makefile	Sun Sep  2 19:54:16 2001
@@ -81,26 +81,20 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DATADIR)/banners
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/banners
 	for file in $(BANNERS); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/banners; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/banners; \
 	done
-	$(INSTALL_DIR) $(DATADIR)/charsets
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/charsets
 	for file in $(CHARSETS); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/charsets; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/charsets; \
 	done
-	$(INSTALL_DIR) $(DATADIR)/data
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/data
 	for file in $(DATAFILES); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/data; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/data; \
 	done
-	-if test x$(PAMDIR) != x; then \
-		$(INSTALL_DIR) $(PAMDIR); \
-		if test -f /lib/security/pam_unix.so; then \
-			$(INSTALL_DATA) cups.suse $(PAMDIR)/cups; \
-		else \
-			$(INSTALL_DATA) cups.pam $(PAMDIR)/cups; \
-		fi \
-	fi
+	$(INSTALL_DIR) $(DESTDIR)$(PAMDIR)
+	$(INSTALL_DATA) cups.pam $(DESTDIR)$(PAMDIR)/cups
 
 
 #
diff -Nru cups-1.1.10/doc/Makefile cups-1.1.10.new/doc/Makefile
--- cups-1.1.10/doc/Makefile	Thu Jul 19 22:17:53 2001
+++ cups-1.1.10.new/doc/Makefile	Sun Sep  2 19:55:11 2001
@@ -118,21 +118,21 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DOCDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(DOCDIR)
 	for file in $(WEBPAGES); do \
-		$(INSTALL_MAN) $$file $(DOCDIR); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR); \
 	done
-	$(INSTALL_MAN) overview.html $(DOCDIR)
-	$(INSTALL_MAN) overview.pdf $(DOCDIR)
+	$(INSTALL_MAN) overview.html $(DESTDIR)$(DOCDIR)
+	$(INSTALL_MAN) overview.pdf $(DESTDIR)$(DOCDIR)
 	for file in $(DOCUMENTS:.shtml=.html); do \
-		$(INSTALL_MAN) $$file $(DOCDIR); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR); \
 	done
 	for file in $(DOCUMENTS:.shtml=.pdf); do \
-		$(INSTALL_MAN) $$file $(DOCDIR); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR); \
 	done
-	$(INSTALL_DIR) $(DOCDIR)/images
+	$(INSTALL_DIR) $(DESTDIR)$(DOCDIR)/images
 	for file in $(WEBIMAGES) $(DOCIMAGES); do \
-		$(INSTALL_MAN) $$file $(DOCDIR)/images; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR)/images; \
 	done
 
 
diff -Nru cups-1.1.10/filter/Makefile cups-1.1.10.new/filter/Makefile
--- cups-1.1.10/filter/Makefile	Fri Aug 17 17:54:02 2001
+++ cups-1.1.10.new/filter/Makefile	Sun Sep  2 19:56:05 2001
@@ -59,19 +59,15 @@
 #
 
 install:
-	$(INSTALL_DIR) $(SERVERBIN)/filter
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/filter
 	for file in $(TARGETS); do \
-		$(INSTALL_BIN) $$file $(SERVERBIN)/filter; \
+		$(INSTALL_BIN) $$file $(DESTDIR)$(SERVERBIN)/filter; \
 	done
-	$(INSTALL_DIR) $(LIBDIR)
-	$(INSTALL_LIB) $(LIBCUPSIMAGE) $(LIBDIR)
-	-if test $(LIBCUPSIMAGE) = "libcupsimage.so.2" -o $(LIBCUPSIMAGE) = "libcupsimage.sl.2"; then \
-		$(RM) $(LIBDIR)/`basename $(LIBCUPSIMAGE) .2`; \
-		$(LN) $(LIBCUPSIMAGE) $(LIBDIR)/`basename $(LIBCUPSIMAGE) .2`; \
-	fi
-	$(INSTALL_DIR) $(INCLUDEDIR)/cups
-	$(INSTALL_DATA) raster.h $(INCLUDEDIR)/cups
-
+	$(INSTALL_DIR) $(DESTDIR)$(LIBDIR)
+	$(INSTALL_LIB) $(LIBCUPSIMAGE) $(DESTDIR)$(LIBDIR)
+	$(LN) $(LIBCUPSIMAGE) $(DESTDIR)$(LIBDIR)/`basename $(LIBCUPSIMAGE) .2`; \
+	$(INSTALL_DIR) $(DESTDIR)$(INCLUDEDIR)/cups
+	$(INSTALL_DATA) raster.h $(DESTDIR)$(INCLUDEDIR)/cups
 
 #
 # formtops
diff -Nru cups-1.1.10/fonts/Makefile cups-1.1.10.new/fonts/Makefile
--- cups-1.1.10/fonts/Makefile	Thu Jul 19 22:17:55 2001
+++ cups-1.1.10.new/fonts/Makefile	Sun Sep  2 20:04:21 2001
@@ -64,9 +64,9 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DATADIR)/fonts
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/fonts
 	for file in $(FONTS); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/fonts; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/fonts; \
 	done
 
 
diff -Nru cups-1.1.10/locale/Makefile cups-1.1.10.new/locale/Makefile
--- cups-1.1.10/locale/Makefile	Wed Jun 27 23:53:15 2001
+++ cups-1.1.10.new/locale/Makefile	Sun Sep  2 19:40:12 2001
@@ -50,10 +50,10 @@
 #
 
 install:
-	$(INSTALL_DIR) $(LOCALEDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(LOCALEDIR)
 	for dir in $(LOCALES) ; do \
-		$(INSTALL_DIR) $(LOCALEDIR)/$$dir ; \
-		$(INSTALL_DATA) $$dir/cups_$$dir $(LOCALEDIR)/$$dir ; \
+		$(INSTALL_DIR) $(DESTDIR)$(LOCALEDIR)/$$dir ; \
+		$(INSTALL_DATA) $$dir/cups_$$dir $(DESTDIR)$(LOCALEDIR)/$$dir ; \
 	done
 
 
diff -Nru cups-1.1.10/man/Makefile cups-1.1.10.new/man/Makefile
--- cups-1.1.10/man/Makefile	Thu Jul 19 22:17:56 2001
+++ cups-1.1.10.new/man/Makefile	Sun Sep  2 19:58:37 2001
@@ -60,43 +60,21 @@
 #
 
 install:
-	$(INSTALL_DIR) $(MANDIR)/man1
+	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man1
 	for file in $(MAN1); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/man1/`basename $$file man`1; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(MANDIR)/man1/`basename $$file man`1; \
 	done
-	$(RM) $(MANDIR)/man1/cancel.1
-	$(LN) lp.1 $(MANDIR)/man1/cancel.1
-	$(INSTALL_DIR) $(MANDIR)/man5
+	echo ".so lp.1" > $(DESTDIR)$(MANDIR)/man1/cancel.1
+	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man5
 	for file in $(MAN5); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/man5/`basename $$file man`5; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(MANDIR)/man5/`basename $$file man`5; \
 	done
-	$(INSTALL_DIR) $(AMANDIR)/man$(MAN8EXT)
+	$(INSTALL_DIR) $(DESTDIR)$(AMANDIR)/man$(MAN8EXT)
 	for file in $(MAN8); do \
-		$(INSTALL_MAN) $$file $(AMANDIR)/man$(MAN8EXT)/`basename $$file man`$(MAN8EXT); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(AMANDIR)/man$(MAN8EXT)/`basename $$file man`$(MAN8EXT); \
 	done
-	$(RM) $(AMANDIR)/man$(MAN8EXT)/reject.$(MAN8EXT)
-	$(LN) accept.$(MAN8EXT) $(AMANDIR)/man$(MAN8EXT)/reject.$(MAN8EXT)
-	$(RM) $(AMANDIR)/man$(MAN8EXT)/disable.$(MAN8EXT)
-	$(LN) enable.$(MAN8EXT) $(AMANDIR)/man$(MAN8EXT)/disable.$(MAN8EXT)
-	$(INSTALL_DIR) $(MANDIR)/cat1
-	for file in $(CAT1); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/cat1; \
-	done
-	$(RM) $(MANDIR)/cat1/cancel.$(CAT1EXT)
-	$(LN) lp.$(CAT1EXT) $(MANDIR)/cat1/cancel.$(CAT1EXT)
-	$(INSTALL_DIR) $(MANDIR)/cat5
-	for file in $(CAT5); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/cat5; \
-	done
-	$(INSTALL_DIR) $(AMANDIR)/cat$(MAN8EXT)
-	for file in $(CAT8); do \
-		$(INSTALL_MAN) $$file $(AMANDIR)/cat$(MAN8EXT); \
-	done
-	$(RM) $(AMANDIR)/cat$(MAN8EXT)/reject.$(CAT8EXT)
-	$(LN) accept.$(CAT8EXT) $(AMANDIR)/cat$(MAN8EXT)/reject.$(CAT8EXT)
-	$(RM) $(AMANDIR)/cat$(MAN8EXT)/disable.$(CAT8EXT)
-	$(LN) enable.$(CAT8EXT) $(AMANDIR)/cat$(MAN8EXT)/disable.$(CAT8EXT)
-
+	echo ".so accept.$(MAN8EXT)" > $(DESTDIR)$(AMANDIR)/man$(MAN8EXT)/reject.$(MAN8EXT)
+	echo ".so enable.$(MAN8EXT)" > $(DESTDIR)$(AMANDIR)/man$(MAN8EXT)/disable.$(MAN8EXT)
 
 #
 # End of "$Id$".
diff -Nru cups-1.1.10/pdftops/Makefile cups-1.1.10.new/pdftops/Makefile
--- cups-1.1.10/pdftops/Makefile	Wed Jun 27 23:53:16 2001
+++ cups-1.1.10.new/pdftops/Makefile	Sun Sep  2 19:40:12 2001
@@ -43,8 +43,8 @@
 #
 
 install:
-	$(INSTALL_DIR) $(SERVERBIN)/filter
-	$(INSTALL_BIN) pdftops $(SERVERBIN)/filter
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/filter
+	$(INSTALL_BIN) pdftops $(DESTDIR)$(SERVERBIN)/filter
 
 
 #
diff -Nru cups-1.1.10/ppd/Makefile cups-1.1.10.new/ppd/Makefile
--- cups-1.1.10/ppd/Makefile	Thu Jul 19 22:17:56 2001
+++ cups-1.1.10.new/ppd/Makefile	Sun Sep  2 19:58:58 2001
@@ -52,9 +52,9 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DATADIR)/model
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/model
 	for file in $(FILES); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/model; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/model; \
 	done
 
 
diff -Nru cups-1.1.10/pstoraster/Makefile cups-1.1.10.new/pstoraster/Makefile
--- cups-1.1.10/pstoraster/Makefile	Thu Jul 19 22:17:57 2001
+++ cups-1.1.10.new/pstoraster/Makefile	Sun Sep  2 19:59:30 2001
@@ -376,11 +376,11 @@
 #
 
 install:	$(TARGETS)
-	$(INSTALL_DIR) $(SERVERBIN)/filter
-	$(INSTALL_BIN) pstoraster $(SERVERBIN)/filter
-	$(INSTALL_DIR) $(DATADIR)/pstoraster
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/filter
+	$(INSTALL_BIN) pstoraster $(DESTDIR)$(SERVERBIN)/filter
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/pstoraster
 	for file in $(DFILES); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/pstoraster; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/pstoraster; \
 	done
 
 
diff -Nru cups-1.1.10/scheduler/Makefile cups-1.1.10.new/scheduler/Makefile
--- cups-1.1.10/scheduler/Makefile	Mon Jul 23 20:48:52 2001
+++ cups-1.1.10.new/scheduler/Makefile	Sun Sep  2 20:00:44 2001
@@ -52,17 +52,17 @@
 #
 
 install:
-	$(INSTALL_DIR) $(SBINDIR)
-	$(INSTALL_BIN) cupsd $(SBINDIR)
-	$(INSTALL_DIR) $(SERVERBIN)/daemon
-	$(INSTALL_BIN) cups-lpd $(SERVERBIN)/daemon
-	$(INSTALL_BIN) cups-polld $(SERVERBIN)/daemon
-	$(INSTALL_DIR) -m 711 -o $(CUPS_USER) -g $(CUPS_GROUP) $(SERVERROOT)/certs
-	$(INSTALL_DIR) $(SERVERROOT)/interfaces
-	$(INSTALL_DIR) $(SERVERROOT)/ppd
-	$(INSTALL_DIR) $(LOGDIR)
-	$(INSTALL_DIR) -m 700 -o $(CUPS_USER) -g $(CUPS_GROUP) $(REQUESTS)
-	$(INSTALL_DIR) -m 1700 -o $(CUPS_USER) -g $(CUPS_GROUP) $(REQUESTS)/tmp
+	$(INSTALL_DIR) $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) cupsd $(DESTDIR)$(SBINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/daemon
+	$(INSTALL_BIN) cups-lpd $(DESTDIR)$(SERVERBIN)/daemon
+	$(INSTALL_BIN) cups-polld $(DESTDIR)$(SERVERBIN)/daemon
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERROOT)/certs
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERROOT)/interfaces
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERROOT)/ppd
+	$(INSTALL_DIR) $(DESTDIR)$(LOGDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(REQUESTS)
+	$(INSTALL_DIR) $(DESTDIR)$(REQUESTS)/tmp
 
 
 #
diff -Nru cups-1.1.10/systemv/Makefile cups-1.1.10.new/systemv/Makefile
--- cups-1.1.10/systemv/Makefile	Fri Aug 17 16:22:55 2001
+++ cups-1.1.10.new/systemv/Makefile	Sun Sep  2 20:01:49 2001
@@ -50,22 +50,19 @@
 #
 
 install:
-	$(INSTALL_DIR) $(BINDIR)
-	$(INSTALL_BIN) accept $(SBINDIR)
-	$(RM) $(SBINDIR)/reject
-	$(LN) accept $(SBINDIR)/reject
-	$(INSTALL_BIN) lpadmin $(SBINDIR)
-	$(INSTALL_BIN) lpinfo $(SBINDIR)
-	$(INSTALL_BIN) lpmove $(SBINDIR)
-	$(INSTALL_BIN) cancel $(BINDIR)
-	$(RM) $(BINDIR)/disable
-	$(LN) ../sbin/accept $(BINDIR)/disable
-	$(RM) $(BINDIR)/enable
-	$(LN) ../sbin/accept $(BINDIR)/enable
-	$(INSTALL_BIN) lp $(BINDIR)
-	$(INSTALL_BIN) lpoptions $(BINDIR)
-	$(INSTALL_BIN) lpstat $(BINDIR)
-	$(INSTALL_BIN) -m 4755 -o $(CUPS_USER) -g $(CUPS_GROUP) lppasswd $(BINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) accept $(DESTDIR)$(SBINDIR)
+	$(LN) -sf accept $(DESTDIR)$(SBINDIR)/reject
+	$(INSTALL_BIN) lpadmin $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) lpinfo $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) lpmove $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) cancel $(DESTDIR)$(BINDIR)
+	$(LN) -sf ../sbin/accept $(DESTDIR)$(BINDIR)/disable
+	$(LN) -sf ../sbin/accept $(DESTDIR)$(BINDIR)/enable
+	$(INSTALL_BIN) lp $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lpoptions $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lpstat $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lppasswd $(DESTDIR)$(BINDIR)
 
 
 #
diff -Nru cups-1.1.10/templates/Makefile cups-1.1.10.new/templates/Makefile
--- cups-1.1.10/templates/Makefile	Fri Aug 17 16:12:12 2001
+++ cups-1.1.10.new/templates/Makefile	Sun Sep  2 20:02:06 2001
@@ -91,9 +91,9 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DATADIR)/templates
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/templates
 	for file in $(FILES); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/templates; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/templates; \
 	done
 
 
