--- cups-1.2.7/filter/pstops.c.orig	2006-11-14 17:54:50.000000000 +0100
+++ cups-1.2.7/filter/pstops.c	2006-12-18 02:33:43.540429154 +0100
@@ -909,23 +909,26 @@
       * for the JCL options...
       */
 
-      if (!doc->saw_eof)
-	puts("%%EOF");
-
-      if (doc->emit_jcl)
+      if (number)
       {
-	if (ppd && ppd->jcl_end)
-	  ppdEmitJCLEnd(ppd, stdout);
-	else
-	  putchar(0x04);
+	if (!doc->saw_eof)
+	  puts("%%EOF");
 
-        ppdEmitJCL(ppd, stdout, doc->job_id, doc->user, doc->title);
-      }
+	if (doc->emit_jcl)
+	{
+	  if (ppd && ppd->jcl_end)
+	    ppdEmitJCLEnd(ppd, stdout);
+	  else
+	    putchar(0x04);
 
-      puts("%!PS-Adobe-3.0");
-      puts("%%Pages: (atend)");
-      puts("%%BoundingBox: (atend)");
-      puts("%%EndComments");
+          ppdEmitJCL(ppd, stdout, doc->job_id, doc->user, doc->title);
+	}
+
+	puts("%!PS-Adobe-3.0");
+	puts("%%Pages: (atend)");
+	puts("%%BoundingBox: (atend)");
+	puts("%%EndComments");
+      }
 
      /*
       * Then copy all of the pages...
