--- cups-1.4b3/config-scripts/cups-directories.m4~	2009-04-13 01:04:51.000000000 +0200
+++ cups-1.4b3/config-scripts/cups-directories.m4	2009-06-01 10:44:33.000000000 +0200
@@ -411,16 +411,13 @@
 AC_SUBST(CUPS_SERVERROOT)
 
 # Transient run-time state
-case "$uname" in
-	Darwin*)
-		# Darwin (Mac OS X)
-		CUPS_STATEDIR="$CUPS_SERVERROOT"
-		;;
-	*)
-		# All others
-		CUPS_STATEDIR="$localstatedir/run/cups"
-		;;
-esac
+AC_ARG_WITH(logdir, [  --with-statedir           set path for certs state],statedir="$withval",statedir="")
+
+if test x$logdir = x; then
+       CUPS_STATEDIR="$localstatedir/lib/cups"
+else
+       CUPS_STATEDIR="$statedir"
+fi
 AC_DEFINE_UNQUOTED(CUPS_STATEDIR, "$CUPS_STATEDIR")
 AC_SUBST(CUPS_STATEDIR)
 
--- cups-1.4b3/config-scripts/cups-network.m4~	2008-09-09 00:03:01.000000000 +0200
+++ cups-1.4b3/config-scripts/cups-network.m4	2009-06-01 10:45:32.000000000 +0200
@@ -49,7 +49,7 @@
 				;;
 			*)
 				# All others use FHS standard...
-				CUPS_DEFAULT_DOMAINSOCKET="$CUPS_STATEDIR/cups.sock"
+				CUPS_DEFAULT_DOMAINSOCKET="$localstatedir/run/cups/cups.sock"
 				;;
 		esac
 	else
