diff -Nur cups-1.1.4.org/scheduler/Makefile cups-1.1.4/scheduler/Makefile
--- cups-1.1.4.org/scheduler/Makefile	Thu Sep 14 17:36:25 2000
+++ cups-1.1.4/scheduler/Makefile	Thu Dec  7 09:16:04 2000
@@ -68,10 +68,8 @@
 	$(CHMOD) ugo+rx $(LOGDIR)
 	-$(MKDIR) $(REQUESTS)
 	$(CHMOD) u+rwx,go-rwx $(REQUESTS)
-	$(CHOWN) lp $(REQUESTS)
 	-$(MKDIR) $(REQUESTS)/tmp
 	$(CHMOD) u+rwx,go-rwx,+t $(REQUESTS)/tmp
-	$(CHOWN) lp $(REQUESTS)/tmp
 
 
 #
