--- cups-1.2.1/templates/pl/jobs-header.tmpl.orig	2006-05-15 22:06:46.000000000 +0200
+++ cups-1.2.1/templates/pl/jobs-header.tmpl	2006-05-27 06:31:18.000000000 +0200
@@ -1,15 +1,15 @@
 <P>{?which_jobs=?<A
-HREF="{?printer_name=?/zadania:{printer_uri_supported}}?which_jobs=completed"><IMG
+HREF="{?printer_name=?/jobs:{printer_uri_supported}}?which_jobs=completed"><IMG
 SRC="/images/button-show-completed.gif" CLASS="button" ALT="Wyświetl zakończone zadania"></A>
-<A HREF="{?printer_name=?/zadania:{printer_uri_supported}}?which_jobs=all"><IMG
+<A HREF="{?printer_name=?/jobs:{printer_uri_supported}}?which_jobs=all"><IMG
 SRC="/images/button-show-all.gif" CLASS="button" ALT="Wyświetl wszystkie zadania">:{which_jobs=all?<A
-HREF="{?printer_name=?/zadania:{printer_uri_supported}}?which_jobs=completed"><IMG
+HREF="{?printer_name=?/jobs:{printer_uri_supported}}?which_jobs=completed"><IMG
 SRC="/images/button-show-completed.gif" CLASS="button" ALT="Wyświetl zakończone zadania"></A>
-<A HREF="{?printer_name=?/zadania:{printer_uri_supported}}"><IMG
+<A HREF="{?printer_name=?/jobs:{printer_uri_supported}}"><IMG
 SRC="/images/button-show-active.gif" CLASS="button" ALT="Wyświetl aktywne zadania">:<A
-HREF="{?printer_name=?/zadania:{printer_uri_supported}}"><IMG
+HREF="{?printer_name=?/jobs:{printer_uri_supported}}"><IMG
 SRC="/images/button-show-active.gif" CLASS="button" ALT="Wyświetl aktywne zadania"></A>
-<A HREF="{?printer_name=?/zadania:{printer_uri_supported}}?which_jobs=all"><IMG
+<A HREF="{?printer_name=?/jobs:{printer_uri_supported}}?which_jobs=all"><IMG
 SRC="/images/button-show-all.gif" CLASS="button" ALT="Wyświetl wszystkie zadania">}}</A></P>
 
 <P ALIGN="CENTER">{total=0?Brak zadań:Wyświetlanie {#job_id} z {total} {?which_jobs=?active:{which_jobs=all?:zakończonych}} zadań{total=1?:}}.</P>
