--- cups-1.5rc1/config-scripts/cups-gssapi.m4~	2011-05-12 07:21:56.000000000 +0200
+++ cups-1.5rc1/config-scripts/cups-gssapi.m4	2011-06-15 21:40:54.182690659 +0200
@@ -46,9 +46,9 @@
 	    AC_MSG_ERROR([--enable-gssapi specified but krb5-config cannot be found.])
 	], [
 	    AC_DEFINE([HAVE_GSSAPI], [1], [Is GSSAPI available?])
-	    CFLAGS="$($KRB5CONFIG --cflags gssapi) $CFLAGS"
-	    CPPFLAGS="$($KRB5CONFIG --cflags gssapi) $CPPFLAGS"
-	    LIBGSSAPI="$($KRB5CONFIG --libs gssapi)"
+	    CFLAGS="$(CFLAGS= LDFLAGS= $KRB5CONFIG --cflags gssapi) $CFLAGS"
+	    CPPFLAGS="$(CFLAGS= LDFLAGS= $KRB5CONFIG --cflags gssapi) $CPPFLAGS"
+	    LIBGSSAPI="$(CFLAGS= LDFLAGS= $KRB5CONFIG --libs gssapi)"
 	])
     ])
 
