--- cups-1.4b3/Makedefs.in~	2009-06-01 10:47:46.000000000 +0200
+++ cups-1.4b3/Makedefs.in	2009-06-01 10:49:03.000000000 +0200
@@ -123,7 +123,7 @@
 BACKLIBS	=	@BACKLIBS@
 BANNERTOPS	=	@BANNERTOPS@
 CFLAGS		=	@CPPFLAGS@ @CFLAGS@
-COMMONLIBS	=	@LIBS@
+COMMONLIBS	=	@LIBS@ $(DNSSDLIBS)
 CUPSDLIBS	=	@CUPSDLIBS@
 CXXFLAGS	=	@CPPFLAGS@ @CXXFLAGS@
 CXXLIBS		=	@CXXLIBS@
--- cups-1.4b3/scheduler/dirsvc.c~	2009-05-14 19:54:37.000000000 +0200
+++ cups-1.4b3/scheduler/dirsvc.c	2009-06-01 10:54:49.000000000 +0200
@@ -2042,7 +2042,7 @@
 
     WebIFRef = DNSSDRef;
     if ((error = DNSServiceRegister(&WebIFRef,
-				    kDNSServiceFlagsShareConnection,
+				    0,
 				    0, webif, "_http._tcp", NULL,
 				    NULL, htons(DNSSDPort), 7,
 				    "\006path=/", dnssdRegisterCallback,
@@ -2765,7 +2765,7 @@
     do
     {
       p->ipp_ref = DNSSDRef;
-      if ((se = DNSServiceRegister(&p->ipp_ref, kDNSServiceFlagsShareConnection,
+      if ((se = DNSServiceRegister(&p->ipp_ref, 0,
                                    0, name, regtype, NULL, NULL,
 				   htons(DNSSDPort), ipp_len, ipp_txt,
 				   dnssdRegisterCallback,
@@ -2854,7 +2854,7 @@
 
       p->printer_ref = DNSSDRef;
       if ((se = DNSServiceRegister(&p->printer_ref,
-                                   kDNSServiceFlagsShareConnection,
+                                   0,
 				   0, name, "_printer._tcp", NULL, NULL,
 				   htons(515), printer_len, printer_txt,
 				   dnssdRegisterCallback,
