diff -urNp -x '*.orig' cups-2.3.3.org/conf/cupsd.conf.in cups-2.3.3/conf/cupsd.conf.in
--- cups-2.3.3.org/conf/cupsd.conf.in	2020-04-27 20:04:29.000000000 +0200
+++ cups-2.3.3/conf/cupsd.conf.in	2021-04-20 22:45:47.956492610 +0200
@@ -3,6 +3,9 @@
 # complete description of this file.
 #
 
+MaxLogSize 0
+Printcap /etc/printcap
+
 # Log general information in error_log - change "@CUPS_LOG_LEVEL@" to "debug"
 # for troubleshooting...
 LogLevel @CUPS_LOG_LEVEL@
