diff -urNp -x '*.orig' cups-2.3.3.org/cups/http-addr.c cups-2.3.3/cups/http-addr.c
--- cups-2.3.3.org/cups/http-addr.c	2020-04-27 20:04:29.000000000 +0200
+++ cups-2.3.3/cups/http-addr.c	2021-04-20 22:45:56.326486636 +0200
@@ -388,7 +388,8 @@ httpAddrLookup(
 
     if (error)
     {
-      if (error == EAI_FAIL)
+      if (error == EAI_FAIL || error == EAI_AGAIN || error == EAI_NODATA ||
+          error == EAI_NONAME)
         cg->need_res_init = 1;
 
       return (httpAddrString(addr, name, namelen));
diff -urNp -x '*.orig' cups-2.3.3.org/cups/http-addrlist.c cups-2.3.3/cups/http-addrlist.c
--- cups-2.3.3.org/cups/http-addrlist.c	2020-04-27 20:04:29.000000000 +0200
+++ cups-2.3.3/cups/http-addrlist.c	2021-04-20 22:45:56.326486636 +0200
@@ -649,7 +649,8 @@ httpAddrGetList(const char *hostname,	/*
     }
     else
     {
-      if (error == EAI_FAIL)
+      if (error == EAI_FAIL || error == EAI_AGAIN || error == EAI_NODATA ||
+          error == EAI_NONAME)
         cg->need_res_init = 1;
 
 #  ifdef _WIN32 /* Really, Microsoft?!? */
