--- cups-2.3.1/man/Makefile~	2019-12-13 15:44:10.000000000 +0100
+++ cups-2.3.1/man/Makefile	2020-02-08 14:28:19.482255123 +0100
@@ -128,19 +128,19 @@ install-data: all
 		$(INSTALL_MAN) $$file $(MANDIR)/man7; \
 	done
 	$(RM) $(MANDIR)/man7/ippeveps.7
-	$(LN) ippevepcl.7 $(MANDIR)/man7/ippeveps.7
+	echo ".so ippevepcl.7" > $(MANDIR)/man7/ippeveps.7
 	echo Installing man pages in $(MANDIR)/man8...
 	$(INSTALL_DIR) -m 755 $(MANDIR)/man8
 	for file in $(MAN8); do \
 		$(INSTALL_MAN) $$file $(MANDIR)/man8; \
 	done
 	$(RM) $(MANDIR)/man8/cupsdisable.8
-	$(LN) cupsenable.8 $(MANDIR)/man8/cupsdisable.8
+	echo ".so cupsenable.8" > $(MANDIR)/man8/cupsdisable.8
 	$(RM) $(MANDIR)/man8/cupsreject.8
-	$(LN) cupsaccept.8 $(MANDIR)/man8/cupsreject.8
+	echo ".so cupsaccept.8" > $(MANDIR)/man8/cupsreject.8
 	for file in cups-deviced.8 cups-driverd.8 cups-exec.8; do \
 		$(RM) $(MANDIR)/man8/$$file; \
-		$(LN) cupsd-helper.8 $(MANDIR)/man8/$$file; \
+		echo ".so cupsd-helper.8" > $(MANDIR)/man8/$$file; \
 	done
 
 

