--- cups-1.1.17/backend/Makefile.wiget	Tue Oct 15 15:59:05 2002
+++ cups-1.1.17/backend/Makefile	Sun Dec  8 23:03:27 2002
@@ -59,12 +59,12 @@
 #
 
 install:	all
-	$(INSTALL_DIR) $(SERVERBIN)/backend
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/backend
 	for file in $(BACKENDS); do \
-		$(INSTALL_BIN) $$file $(SERVERBIN)/backend; \
+		$(INSTALL_BIN) $$file $(DESTDIR)$(SERVERBIN)/backend; \
 	done
-	$(RM) $(SERVERBIN)/backend/http
-	$(LN) ipp $(SERVERBIN)/backend/http
+	$(RM) $(DESTDIR)$(SERVERBIN)/backend/http
+	$(LN) ipp $(DESTDIR)$(SERVERBIN)/backend/http
 
 
 #
--- cups-1.1.17/berkeley/Makefile.wiget	Fri Mar  8 20:18:47 2002
+++ cups-1.1.17/berkeley/Makefile	Sun Dec  8 23:03:27 2002
@@ -56,12 +56,12 @@
 #
 
 install:	all
-	$(INSTALL_DIR) $(BINDIR)
-	$(INSTALL_BIN) lpq $(BINDIR)
-	$(INSTALL_BIN) lpr $(BINDIR)
-	$(INSTALL_BIN) lprm $(BINDIR)
-	$(INSTALL_DIR) $(SBINDIR)
-	$(INSTALL_BIN) lpc $(SBINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lpq $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lpr $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lprm $(DESTDIR)$(BINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) lpc $(DESTDIR)$(SBINDIR)
 
 
 #
--- cups-1.1.17/cgi-bin/Makefile.wiget	Fri Mar  8 20:18:47 2002
+++ cups-1.1.17/cgi-bin/Makefile	Sun Dec  8 23:03:27 2002
@@ -58,9 +58,9 @@
 #
 
 install:	all
-	$(INSTALL_DIR) $(SERVERBIN)/cgi-bin
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/cgi-bin
 	for file in $(CGIS); do \
-		$(INSTALL_BIN) $$file $(SERVERBIN)/cgi-bin; \
+		$(INSTALL_BIN) $$file $(DESTDIR)$(SERVERBIN)/cgi-bin; \
 	done
 
 
--- cups-1.1.17/conf/Makefile.wiget	Fri Aug 30 22:56:43 2002
+++ cups-1.1.17/conf/Makefile	Sun Dec  8 23:03:27 2002
@@ -51,23 +51,23 @@
 #
 
 install:
-	$(INSTALL_DIR) $(SERVERROOT)
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERROOT)
 	for file in $(KEEP); do \
-		if test -r $(SERVERROOT)/$$file ; then \
-			$(INSTALL_DATA) $$file $(SERVERROOT)/$$file.N ; \
+		if test -r $(DESTDIR)$(SERVERROOT)/$$file ; then \
+			$(INSTALL_DATA) $$file $(DESTDIR)$(SERVERROOT)/$$file.N ; \
 		else \
-			$(INSTALL_DATA) $$file $(SERVERROOT) ; \
+			$(INSTALL_DATA) $$file $(DESTDIR)$(SERVERROOT) ; \
 		fi ; \
 	done
 	for file in $(REPLACE); do \
-		if test -r $(SERVERROOT)/$$file ; then \
-			$(MV) $(SERVERROOT)/$$file $(SERVERROOT)/$$file.O ; \
+		if test -r $(DESTDIR)$(SERVERROOT)/$$file ; then \
+			$(MV) $(DESTDIR)$(SERVERROOT)/$$file $(DESTDIR)$(SERVERROOT)/$$file.O ; \
 		fi ; \
-		$(INSTALL_DATA) $$file $(SERVERROOT) ; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(SERVERROOT) ; \
 	done
 	-if test x$(PAMDIR) != x$(BUILDROOT); then \
-		$(INSTALL_DIR) $(PAMDIR); \
-		$(INSTALL_DATA) pam.conf $(PAMDIR)/cups; \
+		$(INSTALL_DIR) $(DESTDIR)$(PAMDIR); \
+		$(INSTALL_DATA) pam.conf $(DESTDIR)$(PAMDIR)/cups; \
 	fi
 
 
--- cups-1.1.17/cups/Makefile.wiget	Thu Sep  5 23:11:16 2002
+++ cups-1.1.17/cups/Makefile	Sun Dec  8 23:03:27 2002
@@ -79,26 +79,26 @@
 #
 
 install:	all	installhdrs
-	$(INSTALL_DIR) $(LIBDIR)
-	$(INSTALL_LIB) $(LIBCUPS) $(LIBDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(LIBDIR)
+	$(INSTALL_LIB) $(LIBCUPS) $(DESTDIR)$(LIBDIR)
 	if test $(LIBCUPS) = "libcups.so.2" -o $(LIBCUPS) = "libcups.sl.2"; then \
-		$(RM) $(LIBDIR)/`basename $(LIBCUPS) .2`; \
-		$(LN) $(LIBCUPS) $(LIBDIR)/`basename $(LIBCUPS) .2`; \
+		$(RM) $(DESTDIR)$(LIBDIR)/`basename $(LIBCUPS) .2`; \
+		$(LN) $(LIBCUPS) $(DESTDIR)$(LIBDIR)/`basename $(LIBCUPS) .2`; \
 	fi
 	if test $(LIBCUPS) = "libcups.2.dylib"; then \
-		$(STRIP) -x $(LIBDIR)/$(LIBCUPS); \
-		$(RM) $(LIBDIR)/libcups.dylib; \
-		$(LN) $(LIBCUPS) $(LIBDIR)/libcups.dylib; \
+		$(STRIP) -x $(DESTDIR)$(LIBDIR)/$(LIBCUPS); \
+		$(RM) $(DESTDIR)$(LIBDIR)/libcups.dylib; \
+		$(LN) $(LIBCUPS) $(DESTDIR)$(LIBDIR)/libcups.dylib; \
 	fi
 	if test $(LIBCUPS) != "libcups.a"; then \
-		$(INSTALL_LIB) libcups.a $(LIBDIR); \
+		$(INSTALL_LIB) libcups.a $(DESTDIR)$(LIBDIR); \
 	fi
-	$(RANLIB) $(LIBDIR)/libcups.a
+	$(RANLIB) $(DESTDIR)$(LIBDIR)/libcups.a
 
 installhdrs:
-	$(INSTALL_DIR) $(INCLUDEDIR)/cups
+	$(INSTALL_DIR) $(DESTDIR)$(INCLUDEDIR)/cups
 	for file in $(HEADERS); do \
-		$(INSTALL_DATA) $$file $(INCLUDEDIR)/cups; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(INCLUDEDIR)/cups; \
 	done
 
 
--- cups-1.1.17/data/Makefile.wiget	Fri Aug 30 22:56:44 2002
+++ cups-1.1.17/data/Makefile	Sun Dec  8 23:03:27 2002
@@ -85,17 +85,17 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DATADIR)/banners
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/banners
 	for file in $(BANNERS); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/banners; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/banners; \
 	done
-	$(INSTALL_DIR) $(DATADIR)/charsets
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/charsets
 	for file in $(CHARSETS); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/charsets; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/charsets; \
 	done
-	$(INSTALL_DIR) $(DATADIR)/data
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/data
 	for file in $(DATAFILES); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/data; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/data; \
 	done
 
 
--- cups-1.1.17/doc/fr/Makefile.wiget	Tue Oct 15 15:54:44 2002
+++ cups-1.1.17/doc/fr/Makefile	Sun Dec  8 23:03:27 2002
@@ -106,21 +106,21 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DOCDIR)/fr
+	$(INSTALL_DIR) $(DESTDIR)$(DOCDIR)/fr
 	for file in $(WEBPAGES); do \
-		$(INSTALL_MAN) $$file $(DOCDIR)/fr; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR)/fr; \
 	done
-	$(INSTALL_MAN) overview.html $(DOCDIR)/fr
-	$(INSTALL_MAN) overview.pdf $(DOCDIR)/fr
+	$(INSTALL_MAN) overview.html $(DESTDIR)$(DOCDIR)/fr
+	$(INSTALL_MAN) overview.pdf $(DESTDIR)$(DOCDIR)/fr
 	for file in $(DOCUMENTS:.shtml=.html); do \
-		$(INSTALL_MAN) $$file $(DOCDIR)/fr; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR)/fr; \
 	done
 	for file in $(DOCUMENTS:.shtml=.pdf); do \
-		$(INSTALL_MAN) $$file $(DOCDIR)/fr; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR)/fr; \
 	done
-	$(INSTALL_DIR) $(DOCDIR)/fr/images
+	$(INSTALL_DIR) $(DESTDIR)$(DOCDIR)/fr/images
 	for file in $(WEBIMAGES); do \
-		$(INSTALL_MAN) $$file $(DOCDIR)/fr/images; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR)/fr/images; \
 	done
 
 
--- cups-1.1.17/doc/Makefile.wiget	Mon Oct 21 01:34:54 2002
+++ cups-1.1.17/doc/Makefile	Sun Dec  8 23:03:27 2002
@@ -123,21 +123,21 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DOCDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(DOCDIR)
 	for file in $(WEBPAGES); do \
-		$(INSTALL_MAN) $$file $(DOCDIR); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR); \
 	done
-	$(INSTALL_MAN) overview.html $(DOCDIR)
-	$(INSTALL_MAN) overview.pdf $(DOCDIR)
+	$(INSTALL_MAN) overview.html $(DESTDIR)$(DOCDIR)
+	$(INSTALL_MAN) overview.pdf $(DESTDIR)$(DOCDIR)
 	for file in $(DOCUMENTS:.shtml=.html); do \
-		$(INSTALL_MAN) $$file $(DOCDIR); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR); \
 	done
 	for file in $(DOCUMENTS:.shtml=.pdf); do \
-		$(INSTALL_MAN) $$file $(DOCDIR); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR); \
 	done
-	$(INSTALL_DIR) $(DOCDIR)/images
+	$(INSTALL_DIR) $(DESTDIR)$(DOCDIR)/images
 	for file in $(WEBIMAGES) $(DOCIMAGES); do \
-		$(INSTALL_MAN) $$file $(DOCDIR)/images; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR)/images; \
 	done
 	cd fr; $(MAKE) $(MFLAGS) install
 
--- cups-1.1.17/filter/Makefile.wiget	Tue Oct 22 20:43:45 2002
+++ cups-1.1.17/filter/Makefile	Sun Dec  8 23:03:27 2002
@@ -70,30 +70,30 @@
 #
 
 install:	all installhdrs
-	$(INSTALL_DIR) $(SERVERBIN)/filter
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/filter
 	for file in $(TARGETS); do \
-		$(INSTALL_BIN) $$file $(SERVERBIN)/filter; \
+		$(INSTALL_BIN) $$file $(DESTDIR)$(SERVERBIN)/filter; \
 	done
-	$(INSTALL_DIR) $(LIBDIR)
-	$(INSTALL_LIB) $(LIBCUPSIMAGE) $(LIBDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(LIBDIR)
+	$(INSTALL_LIB) $(LIBCUPSIMAGE) $(DESTDIR)$(LIBDIR)
 	-if test $(LIBCUPSIMAGE) = "libcupsimage.so.2" -o $(LIBCUPSIMAGE) = "libcupsimage.sl.2"; then \
-		$(RM) $(LIBDIR)/`basename $(LIBCUPSIMAGE) .2`; \
-		$(LN) $(LIBCUPSIMAGE) $(LIBDIR)/`basename $(LIBCUPSIMAGE) .2`; \
+		$(RM) $(DESTDIR)$(LIBDIR)/`basename $(LIBCUPSIMAGE) .2`; \
+		$(LN) $(LIBCUPSIMAGE) $(DESTDIR)$(LIBDIR)/`basename $(LIBCUPSIMAGE) .2`; \
 	fi
 	-if test $(LIBCUPSIMAGE) = "libcupsimage.2.dylib"; then \
-		$(STRIP) -x $(LIBDIR)/$(LIBCUPSIMAGE); \
-		$(RM) $(LIBDIR)/libcupsimage.dylib; \
-		$(LN) $(LIBCUPSIMAGE) $(LIBDIR)/libcupsimage.dylib; \
+		$(STRIP) -x $(DESTDIR)$(LIBDIR)/$(LIBCUPSIMAGE); \
+		$(RM) $(DESTDIR)$(LIBDIR)/libcupsimage.dylib; \
+		$(LN) $(LIBCUPSIMAGE) $(DESTDIR)$(LIBDIR)/libcupsimage.dylib; \
 	fi
 	-if test $(LIBCUPSIMAGE) != "libcupsimage.a"; then \
-		$(INSTALL_LIB) libcupsimage.a $(LIBDIR); \
+		$(INSTALL_LIB) libcupsimage.a $(DESTDIR)$(LIBDIR); \
 	fi
-	$(RANLIB) $(LIBDIR)/libcupsimage.a
+	$(RANLIB) $(DESTDIR)$(LIBDIR)/libcupsimage.a
 
 installhdrs:
-	$(INSTALL_DIR) $(INCLUDEDIR)/cups
-	$(INSTALL_DATA) image.h $(INCLUDEDIR)/cups
-	$(INSTALL_DATA) raster.h $(INCLUDEDIR)/cups
+	$(INSTALL_DIR) $(DESTDIR)$(INCLUDEDIR)/cups
+	$(INSTALL_DATA) image.h $(DESTDIR)$(INCLUDEDIR)/cups
+	$(INSTALL_DATA) raster.h $(DESTDIR)$(INCLUDEDIR)/cups
 
 
 #
--- cups-1.1.17/fonts/Makefile.wiget	Mon Jun 24 22:21:58 2002
+++ cups-1.1.17/fonts/Makefile	Sun Dec  8 23:03:27 2002
@@ -51,9 +51,9 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DATADIR)/fonts
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/fonts
 	for file in $(FONTS); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/fonts; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/fonts; \
 	done
 
 
--- cups-1.1.17/locale/Makefile.wiget	Tue Feb 12 20:37:51 2002
+++ cups-1.1.17/locale/Makefile	Sun Dec  8 23:03:27 2002
@@ -51,10 +51,10 @@
 #
 
 install:
-	$(INSTALL_DIR) $(LOCALEDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(LOCALEDIR)
 	for dir in $(LOCALES) ; do \
-		$(INSTALL_DIR) $(LOCALEDIR)/$$dir ; \
-		$(INSTALL_DATA) $$dir/cups_$$dir $(LOCALEDIR)/$$dir ; \
+		$(INSTALL_DIR) $(DESTDIR)$(LOCALEDIR)/$$dir ; \
+		$(INSTALL_DATA) $$dir/cups_$$dir $(DESTDIR)$(LOCALEDIR)/$$dir ; \
 	done
 
 
--- cups-1.1.17/man/fr/Makefile.wiget	Thu Jun  6 14:31:25 2002
+++ cups-1.1.17/man/fr/Makefile	Sun Dec  8 23:03:27 2002
@@ -60,50 +60,50 @@
 #
 
 install:	all
-	$(INSTALL_DIR) $(MANDIR)/$(LANGUAGE)/man1
+	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/$(LANGUAGE)/man1
 	for file in $(MAN1); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/$(LANGUAGE)/man1/`basename $$file man`1; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(MANDIR)/$(LANGUAGE)/man1/`basename $$file man`1; \
 	done
-	$(RM) $(MANDIR)/$(LANGUAGE)/man1/cancel.1
-	$(LN) lp.1 $(MANDIR)/$(LANGUAGE)/man1/cancel.1
-	$(INSTALL_DIR) $(PMANDIR)/fr/man3
+	$(RM) $(DESTDIR)$(MANDIR)/$(LANGUAGE)/man1/cancel.1
+	$(LN) lp.1 $(DESTDIR)$(MANDIR)/$(LANGUAGE)/man1/cancel.1
+	$(INSTALL_DIR) $(DESTDIR)$(PMANDIR)/fr/man3
 	for file in $(MAN3); do \
-		$(INSTALL_MAN) $$file $(PMANDIR)/fr/man3/`basename $$file man`3; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(PMANDIR)/fr/man3/`basename $$file man`3; \
 	done
-	$(INSTALL_DIR) $(MANDIR)/$(LANGUAGE)/man5
+	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/$(LANGUAGE)/man5
 	for file in $(MAN5); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/$(LANGUAGE)/man5/`basename $$file man`5; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(MANDIR)/$(LANGUAGE)/man5/`basename $$file man`5; \
 	done
-	$(INSTALL_DIR) $(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)
+	$(INSTALL_DIR) $(DESTDIR)$(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)
 	for file in $(MAN8); do \
-		$(INSTALL_MAN) $$file $(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/`basename $$file man`$(MAN8EXT); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/`basename $$file man`$(MAN8EXT); \
 	done
-	$(RM) $(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/reject.$(MAN8EXT)
-	$(LN) accept.$(MAN8EXT) $(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/reject.$(MAN8EXT)
-	$(RM) $(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/disable.$(MAN8EXT)
-	$(LN) enable.$(MAN8EXT) $(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/disable.$(MAN8EXT)
-	$(INSTALL_DIR) $(MANDIR)/$(LANGUAGE)/cat1
+	$(RM) $(DESTDIR)$(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/reject.$(MAN8EXT)
+	$(LN) accept.$(MAN8EXT) $(DESTDIR)$(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/reject.$(MAN8EXT)
+	$(RM) $(DESTDIR)$(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/disable.$(MAN8EXT)
+	$(LN) enable.$(MAN8EXT) $(DESTDIR)$(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/disable.$(MAN8EXT)
+	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/$(LANGUAGE)/cat1
 	for file in $(CAT1); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/$(LANGUAGE)/cat1; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(MANDIR)/$(LANGUAGE)/cat1; \
 	done
-	$(RM) $(MANDIR)/$(LANGUAGE)/cat1/cancel.$(CAT1EXT)
-	$(LN) lp.$(CAT1EXT) $(MANDIR)/$(LANGUAGE)/cat1/cancel.$(CAT1EXT)
-	$(INSTALL_DIR) $(PMANDIR)/fr/cat3
+	$(RM) $(DESTDIR)$(MANDIR)/$(LANGUAGE)/cat1/cancel.$(CAT1EXT)
+	$(LN) lp.$(CAT1EXT) $(DESTDIR)$(MANDIR)/$(LANGUAGE)/cat1/cancel.$(CAT1EXT)
+	$(INSTALL_DIR) $(DESTDIR)$(PMANDIR)/fr/cat3
 	for file in $(CAT3); do \
-		$(INSTALL_MAN) $$file $(PMANDIR)/fr/cat3; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(PMANDIR)/fr/cat3; \
 	done
-	$(INSTALL_DIR) $(MANDIR)/$(LANGUAGE)/cat5
+	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/$(LANGUAGE)/cat5
 	for file in $(CAT5); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/$(LANGUAGE)/cat5; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(MANDIR)/$(LANGUAGE)/cat5; \
 	done
-	$(INSTALL_DIR) $(AMANDIR)/$(LANGUAGE)/cat$(MAN8EXT)
+	$(INSTALL_DIR) $(DESTDIR)$(AMANDIR)/$(LANGUAGE)/cat$(MAN8EXT)
 	for file in $(CAT8); do \
-		$(INSTALL_MAN) $$file $(AMANDIR)/$(LANGUAGE)/cat$(MAN8EXT); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(AMANDIR)/$(LANGUAGE)/cat$(MAN8EXT); \
 	done
-	$(RM) $(AMANDIR)/$(LANGUAGE)/cat$(MAN8EXT)/reject.$(CAT8EXT)
-	$(LN) accept.$(CAT8EXT) $(AMANDIR)/$(LANGUAGE)/cat$(MAN8EXT)/reject.$(CAT8EXT)
-	$(RM) $(AMANDIR)/$(LANGUAGE)/cat$(MAN8EXT)/disable.$(CAT8EXT)
-	$(LN) enable.$(CAT8EXT) $(AMANDIR)/$(LANGUAGE)/cat$(MAN8EXT)/disable.$(CAT8EXT)
+	$(RM) $(DESTDIR)$(AMANDIR)/$(LANGUAGE)/cat$(MAN8EXT)/reject.$(CAT8EXT)
+	$(LN) accept.$(CAT8EXT) $(DESTDIR)$(AMANDIR)/$(LANGUAGE)/cat$(MAN8EXT)/reject.$(CAT8EXT)
+	$(RM) $(DESTDIR)$(AMANDIR)/$(LANGUAGE)/cat$(MAN8EXT)/disable.$(CAT8EXT)
+	$(LN) enable.$(CAT8EXT) $(DESTDIR)$(AMANDIR)/$(LANGUAGE)/cat$(MAN8EXT)/disable.$(CAT8EXT)
 
 
 #
--- cups-1.1.17/man/Makefile.wiget	Sun Oct 20 22:13:53 2002
+++ cups-1.1.17/man/Makefile	Sun Dec  8 23:03:27 2002
@@ -68,50 +68,50 @@
 #
 
 install:	all
-	$(INSTALL_DIR) $(MANDIR)/man1
+	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man1
 	for file in $(MAN1); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/man1/`basename $$file man`1; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(MANDIR)/man1/`basename $$file man`1; \
 	done
-	$(RM) $(MANDIR)/man1/cancel.1
-	$(LN) lp.1 $(MANDIR)/man1/cancel.1
-	$(INSTALL_DIR) $(PMANDIR)/man3
+	$(RM) $(DESTDIR)$(MANDIR)/man1/cancel.1
+	$(LN) lp.1 $(DESTDIR)$(MANDIR)/man1/cancel.1
+	$(INSTALL_DIR) $(DESTDIR)$(PMANDIR)/man3
 	for file in $(MAN3); do \
-		$(INSTALL_MAN) $$file $(PMANDIR)/man3/`basename $$file man`3; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(PMANDIR)/man3/`basename $$file man`3; \
 	done
-	$(INSTALL_DIR) $(MANDIR)/man5
+	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man5
 	for file in $(MAN5); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/man5/`basename $$file man`5; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(MANDIR)/man5/`basename $$file man`5; \
 	done
-	$(INSTALL_DIR) $(AMANDIR)/man$(MAN8EXT)
+	$(INSTALL_DIR) $(DESTDIR)$(AMANDIR)/man$(MAN8EXT)
 	for file in $(MAN8); do \
-		$(INSTALL_MAN) $$file $(AMANDIR)/man$(MAN8EXT)/`basename $$file man`$(MAN8EXT); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(AMANDIR)/man$(MAN8EXT)/`basename $$file man`$(MAN8EXT); \
 	done
-	$(RM) $(AMANDIR)/man$(MAN8EXT)/reject.$(MAN8EXT)
-	$(LN) accept.$(MAN8EXT) $(AMANDIR)/man$(MAN8EXT)/reject.$(MAN8EXT)
-	$(RM) $(AMANDIR)/man$(MAN8EXT)/disable.$(MAN8EXT)
-	$(LN) enable.$(MAN8EXT) $(AMANDIR)/man$(MAN8EXT)/disable.$(MAN8EXT)
-	$(INSTALL_DIR) $(MANDIR)/cat1
+	$(RM) $(DESTDIR)$(AMANDIR)/man$(MAN8EXT)/reject.$(MAN8EXT)
+	$(LN) accept.$(MAN8EXT) $(DESTDIR)$(AMANDIR)/man$(MAN8EXT)/reject.$(MAN8EXT)
+	$(RM) $(DESTDIR)$(AMANDIR)/man$(MAN8EXT)/disable.$(MAN8EXT)
+	$(LN) enable.$(MAN8EXT) $(DESTDIR)$(AMANDIR)/man$(MAN8EXT)/disable.$(MAN8EXT)
+	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/cat1
 	for file in $(CAT1); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/cat1; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(MANDIR)/cat1; \
 	done
-	$(RM) $(MANDIR)/cat1/cancel.$(CAT1EXT)
-	$(LN) lp.$(CAT1EXT) $(MANDIR)/cat1/cancel.$(CAT1EXT)
-	$(INSTALL_DIR) $(PMANDIR)/cat3
+	$(RM) $(DESTDIR)$(MANDIR)/cat1/cancel.$(CAT1EXT)
+	$(LN) lp.$(CAT1EXT) $(DESTDIR)$(MANDIR)/cat1/cancel.$(CAT1EXT)
+	$(INSTALL_DIR) $(DESTDIR)$(PMANDIR)/cat3
 	for file in $(CAT3); do \
-		$(INSTALL_MAN) $$file $(PMANDIR)/cat3; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(PMANDIR)/cat3; \
 	done
-	$(INSTALL_DIR) $(MANDIR)/cat5
+	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/cat5
 	for file in $(CAT5); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/cat5; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(MANDIR)/cat5; \
 	done
-	$(INSTALL_DIR) $(AMANDIR)/cat$(MAN8EXT)
+	$(INSTALL_DIR) $(DESTDIR)$(AMANDIR)/cat$(MAN8EXT)
 	for file in $(CAT8); do \
-		$(INSTALL_MAN) $$file $(AMANDIR)/cat$(MAN8EXT); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(AMANDIR)/cat$(MAN8EXT); \
 	done
-	$(RM) $(AMANDIR)/cat$(MAN8EXT)/reject.$(CAT8EXT)
-	$(LN) accept.$(CAT8EXT) $(AMANDIR)/cat$(MAN8EXT)/reject.$(CAT8EXT)
-	$(RM) $(AMANDIR)/cat$(MAN8EXT)/disable.$(CAT8EXT)
-	$(LN) enable.$(CAT8EXT) $(AMANDIR)/cat$(MAN8EXT)/disable.$(CAT8EXT)
+	$(RM) $(DESTDIR)$(AMANDIR)/cat$(MAN8EXT)/reject.$(CAT8EXT)
+	$(LN) accept.$(CAT8EXT) $(DESTDIR)$(AMANDIR)/cat$(MAN8EXT)/reject.$(CAT8EXT)
+	$(RM) $(DESTDIR)$(AMANDIR)/cat$(MAN8EXT)/disable.$(CAT8EXT)
+	$(LN) enable.$(CAT8EXT) $(DESTDIR)$(AMANDIR)/cat$(MAN8EXT)/disable.$(CAT8EXT)
 	for dir in $(LANGDIRS); do \
 		echo "Installing all in man/$$dir..."; \
 		(cd $$dir; $(MAKE) $(MFLAGS) install); \
--- cups-1.1.17/pdftops/Makefile.wiget	Thu Oct 10 01:23:07 2002
+++ cups-1.1.17/pdftops/Makefile	Sun Dec  8 23:03:27 2002
@@ -86,8 +86,8 @@
 #
 
 install:	all
-	$(INSTALL_DIR) $(SERVERBIN)/filter
-	$(INSTALL_BIN) pdftops $(SERVERBIN)/filter
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/filter
+	$(INSTALL_BIN) pdftops $(DESTDIR)$(SERVERBIN)/filter
 
 
 #
--- cups-1.1.17/ppd/Makefile.wiget	Wed Jan  2 18:59:08 2002
+++ cups-1.1.17/ppd/Makefile	Sun Dec  8 23:03:27 2002
@@ -52,9 +52,9 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DATADIR)/model
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/model
 	for file in $(FILES); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/model; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/model; \
 	done
 
 
--- cups-1.1.17/scheduler/Makefile.wiget	Wed Jun  5 16:50:59 2002
+++ cups-1.1.17/scheduler/Makefile	Sun Dec  8 23:03:27 2002
@@ -60,17 +60,17 @@
 #
 
 install:	all
-	$(INSTALL_DIR) $(SBINDIR)
-	$(INSTALL_BIN) cupsd $(SBINDIR)
-	$(INSTALL_DIR) $(SERVERBIN)/daemon
-	$(INSTALL_BIN) cups-lpd $(SERVERBIN)/daemon
-	$(INSTALL_BIN) cups-polld $(SERVERBIN)/daemon
-	$(INSTALL_DIR) -m 711 $(SERVERROOT)/certs
-	$(INSTALL_DIR) $(SERVERROOT)/interfaces
-	$(INSTALL_DIR) $(SERVERROOT)/ppd
-	$(INSTALL_DIR) $(LOGDIR)
-	$(INSTALL_DIR) -m 700 $(REQUESTS)
-	$(INSTALL_DIR) -m 1700 $(REQUESTS)/tmp
+	$(INSTALL_DIR) $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) cupsd $(DESTDIR)$(SBINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/daemon
+	$(INSTALL_BIN) cups-lpd $(DESTDIR)$(SERVERBIN)/daemon
+	$(INSTALL_BIN) cups-polld $(DESTDIR)$(SERVERBIN)/daemon
+	$(INSTALL_DIR) -m 711 $(DESTDIR)$(SERVERROOT)/certs
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERROOT)/interfaces
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERROOT)/ppd
+	$(INSTALL_DIR) $(DESTDIR)$(LOGDIR)
+	$(INSTALL_DIR) -m 700 $(DESTDIR)$(REQUESTS)
+	$(INSTALL_DIR) -m 1700 $(DESTDIR)$(REQUESTS)/tmp
 
 
 #
--- cups-1.1.17/systemv/Makefile.wiget	Wed Jun  5 17:11:58 2002
+++ cups-1.1.17/systemv/Makefile	Sun Dec  8 23:03:27 2002
@@ -58,25 +58,25 @@
 #
 
 install:	all
-	$(INSTALL_DIR) $(SBINDIR)
-	$(INSTALL_BIN) accept $(SBINDIR)
-	$(RM) $(SBINDIR)/reject
-	$(LN) accept $(SBINDIR)/reject
-	$(INSTALL_BIN) cupsaddsmb $(SBINDIR)
-	$(INSTALL_BIN) lpadmin $(SBINDIR)
-	$(INSTALL_BIN) lpinfo $(SBINDIR)
-	$(INSTALL_BIN) lpmove $(SBINDIR)
-	$(INSTALL_DIR) $(BINDIR)
-	$(INSTALL_BIN) cancel $(BINDIR)
-	$(RM) $(BINDIR)/disable
-	$(LN) ../sbin/accept $(BINDIR)/disable
-	$(RM) $(BINDIR)/enable
-	$(LN) ../sbin/accept $(BINDIR)/enable
-	$(INSTALL_BIN) lp $(BINDIR)
-	$(INSTALL_BIN) lpoptions $(BINDIR)
-	$(INSTALL_BIN) lpstat $(BINDIR)
-	$(INSTALL_BIN) lppasswd $(BINDIR)
-	-$(INSTALL_BIN) -m 4755 -o $(CUPS_USER) -g $(CUPS_GROUP) lppasswd $(BINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) accept $(DESTDIR)$(SBINDIR)
+	$(RM) $(DESTDIR)$(SBINDIR)/reject
+	$(LN) accept $(DESTDIR)$(SBINDIR)/reject
+	$(INSTALL_BIN) cupsaddsmb $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) lpadmin $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) lpinfo $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) lpmove $(DESTDIR)$(SBINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) cancel $(DESTDIR)$(BINDIR)
+	$(RM) $(DESTDIR)$(BINDIR)/disable
+	$(LN) ../sbin/accept $(DESTDIR)$(BINDIR)/disable
+	$(RM) $(DESTDIR)$(BINDIR)/enable
+	$(LN) ../sbin/accept $(DESTDIR)$(BINDIR)/enable
+	$(INSTALL_BIN) lp $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lpoptions $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lpstat $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lppasswd $(DESTDIR)$(BINDIR)
+	-$(INSTALL_BIN) -m 4755 -o $(CUPS_USER) -g $(CUPS_GROUP) lppasswd $(DESTDIR)$(BINDIR)
 
 
 #
--- cups-1.1.17/templates/fr/Makefile.wiget	Tue Oct 15 15:54:47 2002
+++ cups-1.1.17/templates/fr/Makefile	Sun Dec  8 23:08:40 2002
@@ -94,9 +94,9 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DATADIR)/templates/fr
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/templates/fr
 	for file in $(FILES); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/templates/fr; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/templates/fr; \
 	done
 
 
--- cups-1.1.17/templates/Makefile.wiget	Sun Oct 20 22:15:02 2002
+++ cups-1.1.17/templates/Makefile	Sun Dec  8 23:03:27 2002
@@ -95,9 +95,9 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DATADIR)/templates
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/templates
 	for file in $(FILES); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/templates; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/templates; \
 	done
 	cd fr; $(MAKE) $(MFLAGS) install
 
--- cups-1.1.17/Makefile.wiget	Tue Apr 23 19:49:34 2002
+++ cups-1.1.17/Makefile	Sun Dec  8 23:03:27 2002
@@ -87,32 +87,8 @@
 	echo Installing in templates...
 	(cd templates; $(MAKE) $(MFLAGS) install)
 	echo Installing cups-config script...
-	$(INSTALL_DIR) $(BINDIR)
-	$(INSTALL_SCRIPT) cups-config $(BINDIR)/cups-config
-	echo Installing startup script...
-	if test "x$(INITDIR)" != "x"; then \
-		$(INSTALL_DIR) $(BUILDROOT)$(INITDIR)/init.d; \
-		$(INSTALL_SCRIPT) cups.sh $(BUILDROOT)$(INITDIR)/init.d/cups; \
-		$(INSTALL_DIR) $(BUILDROOT)$(INITDIR)/rc0.d; \
-		$(INSTALL_SCRIPT) cups.sh  $(BUILDROOT)$(INITDIR)/rc0.d/K00cups; \
-		$(INSTALL_DIR) $(BUILDROOT)$(INITDIR)/rc2.d; \
-		$(INSTALL_SCRIPT) cups.sh $(BUILDROOT)$(INITDIR)/rc2.d/S99cups; \
-		$(INSTALL_DIR) $(BUILDROOT)$(INITDIR)/rc3.d; \
-		$(INSTALL_SCRIPT) cups.sh $(BUILDROOT)$(INITDIR)/rc3.d/S99cups; \
-		$(INSTALL_DIR) $(BUILDROOT)$(INITDIR)/rc5.d; \
-		$(INSTALL_SCRIPT) cups.sh $(BUILDROOT)$(INITDIR)/rc5.d/S99cups; \
-	fi
-	if test "x$(INITDIR)" = "x" -a "x$(INITDDIR)" != "x"; then \
-		$(INSTALL_DIR) $(BUILDROOT)$(INITDDIR); \
-		if test "$(INITDDIR)" = "/System/Library/StartupItems/PrintingServices"; then \
-			$(INSTALL_SCRIPT) cups.sh $(BUILDROOT)$(INITDDIR)/PrintingServices; \
-			$(INSTALL_DATA) cups.plist $(BUILDROOT)$(INITDDIR)/StartupParameters.plist; \
-			$(INSTALL_DIR) $(BUILDROOT)$(INITDDIR)/Resources/English.lproj; \
-			$(INSTALL_DATA) cups.strings $(BUILDROOT)$(INITDDIR)/Resources/English.lproj/Localizable.strings; \
-		else \
-			$(INSTALL_SCRIPT) cups.sh $(BUILDROOT)$(INITDDIR)/cups; \
-		fi \
-	fi
+	$(INSTALL_DIR) $(DESTDIR)$(BINDIR)
+	$(INSTALL_SCRIPT) cups-config $(DESTDIR)$(BINDIR)/cups-config
 
 #
 # Install source and header files...
