diff -urN cups-1.1.20rc4.org/Makedefs.in cups-1.1.20rc4/Makedefs.in
--- cups-1.1.20rc4.org/Makedefs.in	2003-10-19 21:40:50.000000000 +0200
+++ cups-1.1.20rc4/Makedefs.in	2003-10-19 21:41:20.000000000 +0200
@@ -41,13 +41,13 @@
 RM		=	@RM@ -f
 SED		=	@SED@
 SHELL		=	/bin/sh
-STRIP		=	@STRIP@
+STRIP		=	/bin/true
 
 #
 # Installation programs...
 #
 
-INSTALL_BIN	=	$(LIBTOOL) $(INSTALL) -m 755 -s
+INSTALL_BIN	=	$(LIBTOOL) $(INSTALL) -m 755
 INSTALL_DATA	=	$(INSTALL) -m 644
 INSTALL_DIR	=	$(INSTALL) -d
 INSTALL_LIB	=	$(LIBTOOL) $(INSTALL) -m 755
