diff -urN cups-1.1.21.org/man/fr/Makefile cups-1.1.21/man/fr/Makefile
--- cups-1.1.21.org/man/fr/Makefile	2004-09-24 01:17:20.456195732 +0200
+++ cups-1.1.21/man/fr/Makefile	2004-09-24 01:18:25.086350238 +0200
@@ -65,7 +65,7 @@
 		$(INSTALL_MAN) $$file $(MANDIR)/$(LANGUAGE)/man1/`basename $$file man`1; \
 	done
 	$(RM) $(MANDIR)/$(LANGUAGE)/man1/cancel.1
-	$(LN) lp.1 $(MANDIR)/$(LANGUAGE)/man1/cancel.1
+	echo ".so lp.1" > $(MANDIR)/$(LANGUAGE)/man1/cancel.1
 	$(INSTALL_DIR) $(MANDIR)/$(LANGUAGE)/man5
 	for file in $(MAN5); do \
 		$(INSTALL_MAN) $$file $(MANDIR)/$(LANGUAGE)/man5/`basename $$file man`5; \
@@ -75,9 +75,9 @@
 		$(INSTALL_MAN) $$file $(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/`basename $$file man`$(MAN8EXT); \
 	done
 	$(RM) $(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/reject.$(MAN8EXT)
-	$(LN) accept.$(MAN8EXT) $(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/reject.$(MAN8EXT)
+	echo ".so accept.$(MAN8EXT)" > $(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/reject.$(MAN8EXT)
 	$(RM) $(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/disable.$(MAN8EXT)
-	$(LN) enable.$(MAN8EXT) $(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/disable.$(MAN8EXT)
+	echo ".so enable.$(MAN8EXT)" > $(AMANDIR)/$(LANGUAGE)/man$(MAN8EXT)/disable.$(MAN8EXT)
 	$(INSTALL_DIR) $(MANDIR)/$(LANGUAGE)/cat1
 	for file in $(CAT1); do \
 		$(INSTALL_MAN) $$file $(MANDIR)/$(LANGUAGE)/cat1; \
diff -urN cups-1.1.21.org/man/Makefile cups-1.1.21/man/Makefile
--- cups-1.1.21.org/man/Makefile	2004-09-24 01:17:20.635157394 +0200
+++ cups-1.1.21/man/Makefile	2004-09-24 01:18:00.631589097 +0200
@@ -74,7 +74,7 @@
 		$(INSTALL_MAN) $$file $(MANDIR)/man1/`basename $$file man`1; \
 	done
 	$(RM) $(MANDIR)/man1/cancel.1
-	$(LN) lp.1 $(MANDIR)/man1/cancel.1
+	echo ".so lp.1" > $(MANDIR)/man1/cancel.1
 	$(INSTALL_DIR) $(MANDIR)/man5
 	for file in $(MAN5); do \
 		echo Installing $$file in $(MANDIR)/man5...; \
@@ -86,9 +86,9 @@
 		$(INSTALL_MAN) $$file $(AMANDIR)/man$(MAN8EXT)/`basename $$file man`$(MAN8EXT); \
 	done
 	$(RM) $(AMANDIR)/man$(MAN8EXT)/reject.$(MAN8EXT)
-	$(LN) accept.$(MAN8EXT) $(AMANDIR)/man$(MAN8EXT)/reject.$(MAN8EXT)
+	echo ".so accept.$(MAN8EXT)" > $(AMANDIR)/man$(MAN8EXT)/reject.$(MAN8EXT)
 	$(RM) $(AMANDIR)/man$(MAN8EXT)/disable.$(MAN8EXT)
-	$(LN) enable.$(MAN8EXT) $(AMANDIR)/man$(MAN8EXT)/disable.$(MAN8EXT)
+	echo ".so enable.$(MAN8EXT)" > $(AMANDIR)/man$(MAN8EXT)/disable.$(MAN8EXT)
 	$(INSTALL_DIR) $(MANDIR)/cat1
 	for file in $(CAT1); do \
 		echo Installing $$file in $(MANDIR)/cat1...; \
