--- cups-1.4b2/berkeley/lpr.c~	2008-07-12 00:46:21.000000000 +0200
+++ cups-1.4b2/berkeley/lpr.c	2008-12-16 11:53:15.000000000 +0100
@@ -383,7 +383,7 @@
     }
   }
   else if ((job_id = cupsCreateJob(CUPS_HTTP_DEFAULT, printer,
-                                   title ? title : "(stdin)",
+                                   title ? title : "STDIN",
                                    num_options, options)) > 0)
   {
     http_status_t	status;		/* Write status */
--- cups-1.4b2/systemv/lp.c~	2008-11-18 01:42:34.000000000 +0100
+++ cups-1.4b2/systemv/lp.c	2008-12-16 11:54:39.000000000 +0100
@@ -618,7 +618,7 @@
   if (num_files > 0)
     job_id = cupsPrintFiles(printer, num_files, files, title, num_options, options);
   else if ((job_id = cupsCreateJob(CUPS_HTTP_DEFAULT, printer,
-                                   title ? title : "(stdin)",
+                                   title ? title : "STDIN",
                                    num_options, options)) > 0)
   {
     http_status_t	status;		/* Write status */
