Index: cups/request.c
===================================================================
--- cups/request.c	(revision 9973)
+++ cups/request.c	(revision 9974)
@@ -248,16 +248,9 @@
 
       while ((bytes = (int)read(infile, buffer, sizeof(buffer))) > 0)
       {
-	if (httpCheck(http))
-	{
-	  _httpUpdate(http, &status);
-
-	  if (status >= HTTP_MULTIPLE_CHOICES)
-	    break;
-        }
-
-  	if (httpWrite2(http, buffer, bytes) < bytes)
-          break;
+        if ((status = cupsWriteRequestData(http, buffer, bytes))
+                != HTTP_CONTINUE)
+	  break;
       }
     }
 
@@ -265,13 +258,11 @@
     * Get the server's response...
     */
 
-    if (status == HTTP_CONTINUE || status == HTTP_OK)
+    if (status != HTTP_ERROR)
     {
       response = cupsGetResponse(http, resource);
-      status   = http->status;
+      status   = httpGetStatus(http);
     }
-    else
-      httpFlush(http);
 
     DEBUG_printf(("2cupsDoIORequest: status=%d", status));
 
@@ -785,7 +776,17 @@
     */
 
     if (status >= HTTP_MULTIPLE_CHOICES)
+    {
+      _cupsSetHTTPError(status);
+
+      do
+      {
+	status = httpUpdate(http);
+      }
+      while (status != HTTP_ERROR && http->state == HTTP_POST_RECV);
+
       httpFlush(http);
+    }
 
     switch (status)
     {
@@ -922,6 +923,13 @@
       if (status >= HTTP_MULTIPLE_CHOICES)
       {
         _cupsSetHTTPError(status);
+
+	do
+	{
+	  status = httpUpdate(http);
+	}
+	while (status != HTTP_ERROR && http->state == HTTP_POST_RECV);
+
         httpFlush(http);
       }
 
