--- cups-1.1.14/pdftops/FontFile.cxx.orig	2001-12-07 22:02:10.000000000 +0100
+++ cups-1.1.14/pdftops/FontFile.cxx	2005-09-05 13:21:44.000000000 +0200
@@ -15,6 +15,7 @@
 #include <stddef.h>
 #include <string.h>
 #include <ctype.h>
+#include <error.h>
 #include "gmem.h"
 #include "Error.h"
 #include "FontFile.h"
@@ -2483,6 +2484,8 @@
 	  } else {
 	    list[k1] = 2 * getUShort(pos + 2*k1);
 	  }
+	  if (list[k1] < 0 || list[k1] > len)
+	    error (1, 0, "bad loca table pos value");
 	}
 	k1 = 0;
 	while (k1 < nGlyphs) {
