--- cups-1.1.13/Makefile.orig	Thu Jan 24 19:51:04 2002
+++ cups-1.1.13/Makefile	Tue Feb  5 08:20:11 2002
@@ -76,33 +76,8 @@
 	echo Installing in templates...
 	(cd templates; $(MAKE) $(MFLAGS) install)
 	echo Installing cups-config script...
-	$(INSTALL_DIR) $(BINDIR)
-	$(INSTALL_SCRIPT) cups-config $(BINDIR)/cups-config
-	echo Installing startup script...
-	if test "x$(INITDIR)" != "x"; then \
-		$(INSTALL_DIR) $(BUILDROOT)$(INITDIR)/init.d; \
-		$(INSTALL_SCRIPT) cups.sh $(BUILDROOT)$(INITDIR)/init.d/cups; \
-		$(INSTALL_DIR) $(BUILDROOT)$(INITDIR)/rc0.d; \
-		$(INSTALL_SCRIPT) cups.sh  $(BUILDROOT)$(INITDIR)/rc0.d/K00cups; \
-		$(INSTALL_DIR) $(BUILDROOT)$(INITDIR)/rc2.d; \
-		$(INSTALL_SCRIPT) cups.sh $(BUILDROOT)$(INITDIR)/rc2.d/S99cups; \
-		$(INSTALL_DIR) $(BUILDROOT)$(INITDIR)/rc3.d; \
-		$(INSTALL_SCRIPT) cups.sh $(BUILDROOT)$(INITDIR)/rc3.d/S99cups; \
-		$(INSTALL_DIR) $(BUILDROOT)$(INITDIR)/rc5.d; \
-		$(INSTALL_SCRIPT) cups.sh $(BUILDROOT)$(INITDIR)/rc5.d/S99cups; \
-	fi
-	if test "x$(INITDIR)" = "x" -a "x$(INITDDIR)" != "x"; then \
-		$(INSTALL_DIR) $(BUILDROOT)$(INITDDIR); \
-		if test "$(INITDDIR)" = "/System/Library/StartupItems/CUPS"; then \
-			$(INSTALL_SCRIPT) cups.sh $(BUILDROOT)$(INITDDIR)/CUPS; \
-			$(INSTALL_DATA) cups.plist $(BUILDROOT)$(INITDDIR)/StartupParameters.plist; \
-			$(INSTALL_DIR) $(BUILDROOT)$(INITDDIR)/Resources/English.lproj; \
-			$(INSTALL_DATA) cups.strings $(BUILDROOT)$(INITDDIR)/Resources/English.lproj/Localizable.strings; \
-		else \
-			$(INSTALL_SCRIPT) cups.sh $(BUILDROOT)$(INITDDIR)/cups; \
-		fi \
-	fi
-
+	$(INSTALL_DIR) $(DESTDIR)$(bindir)
+	$(INSTALL_SCRIPT) cups-config $(DESTDIR)$(bindir)/cups-config
 
 #
 # Run the test suite...
diff -Nur cups-1.1.12/backend/Makefile cups-1.1.12-new/backend/Makefile
--- cups-1.1.12/backend/Makefile	Tue Jul 24 21:56:41 2001
+++ cups-1.1.12-new/backend/Makefile	Sat Nov 24 20:27:51 2001
@@ -49,12 +49,11 @@
 #
 
 install:
-	$(INSTALL_DIR) $(SERVERBIN)/backend
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/backend
 	for file in $(BACKENDS); do \
-		$(INSTALL_BIN) $$file $(SERVERBIN)/backend; \
+		$(INSTALL_BIN) $$file $(DESTDIR)$(SERVERBIN)/backend; \
 	done
-	$(RM) $(SERVERBIN)/backend/http
-	$(LN) ipp $(SERVERBIN)/backend/http
+	$(LN) -sf ipp $(DESTDIR)$(SERVERBIN)/backend/http
 
 
 #
diff -Nur cups-1.1.12/berkeley/Makefile cups-1.1.12-new/berkeley/Makefile
--- cups-1.1.12/berkeley/Makefile	Thu Jul 19 22:17:51 2001
+++ cups-1.1.12-new/berkeley/Makefile	Sat Nov 24 20:27:51 2001
@@ -48,12 +48,12 @@
 #
 
 install:
-	$(INSTALL_DIR) $(BINDIR)
-	$(INSTALL_BIN) lpq $(BINDIR)
-	$(INSTALL_BIN) lpr $(BINDIR)
-	$(INSTALL_BIN) lprm $(BINDIR)
-	$(INSTALL_DIR) $(SBINDIR)
-	$(INSTALL_BIN) lpc $(SBINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lpq $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lpr $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lprm $(DESTDIR)$(BINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) lpc $(DESTDIR)$(SBINDIR)
 
 
 #
diff -Nur cups-1.1.12/cgi-bin/Makefile cups-1.1.12-new/cgi-bin/Makefile
--- cups-1.1.12/cgi-bin/Makefile	Thu Jul 19 22:17:51 2001
+++ cups-1.1.12-new/cgi-bin/Makefile	Sat Nov 24 20:27:51 2001
@@ -50,9 +50,9 @@
 #
 
 install:
-	$(INSTALL_DIR) $(SERVERBIN)/cgi-bin
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/cgi-bin
 	for file in $(CGIS); do \
-		$(INSTALL_BIN) $$file $(SERVERBIN)/cgi-bin; \
+		$(INSTALL_BIN) $$file $(DESTDIR)$(SERVERBIN)/cgi-bin; \
 	done
 
 
diff -Nur cups-1.1.12/conf/Makefile cups-1.1.12-new/conf/Makefile
--- cups-1.1.12/conf/Makefile	Wed Jun 27 23:53:08 2001
+++ cups-1.1.12-new/conf/Makefile	Sat Nov 24 20:27:51 2001
@@ -51,19 +51,12 @@
 #
 
 install:
-	$(INSTALL_DIR) $(SERVERROOT)
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERROOT)
 	for file in $(KEEP); do \
-		if test -r $(SERVERROOT)/$$file ; then \
-			$(INSTALL_DATA) $$file $(SERVERROOT)/$$file.N ; \
-		else \
-			$(INSTALL_DATA) $$file $(SERVERROOT) ; \
-		fi ; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(SERVERROOT) ; \
 	done
 	for file in $(REPLACE); do \
-		if test -r $(SERVERROOT)/$$file ; then \
-			$(MV) $(SERVERROOT)/$$file $(SERVERROOT)/$$file.O ; \
-		fi ; \
-		$(INSTALL_DATA) $$file $(SERVERROOT) ; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(SERVERROOT) ; \
 	done
 
 
diff -Nur cups-1.1.12/cups/Makefile cups-1.1.12-new/cups/Makefile
--- cups-1.1.12/cups/Makefile	Tue Oct 30 21:37:14 2001
+++ cups-1.1.12-new/cups/Makefile	Sat Nov 24 20:32:04 2001
@@ -68,25 +68,14 @@
 #
 
 install:	all
-	$(INSTALL_DIR) $(INCLUDEDIR)/cups
+	$(INSTALL_DIR) $(DESTDIR)$(INCLUDEDIR)/cups
 	for file in $(HEADERS); do \
-		$(INSTALL_DATA) $$file $(INCLUDEDIR)/cups; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(INCLUDEDIR)/cups; \
 	done
-	$(INSTALL_DIR) $(LIBDIR)
-	$(INSTALL_LIB) $(LIBCUPS) $(LIBDIR)
-	if test $(LIBCUPS) = "libcups.so.2" -o $(LIBCUPS) = "libcups.sl.2"; then \
-		$(INSTALL_LIB) libcups.a $(LIBDIR); \
-		$(RM) $(LIBDIR)/`basename $(LIBCUPS) .2`; \
-		$(LN) $(LIBCUPS) $(LIBDIR)/`basename $(LIBCUPS) .2`; \
-	fi
-	if test $(LIBCUPS) = "libcups.2.dylib"; then \
-		$(INSTALL_LIB) libcups.a $(LIBDIR); \
-		$(RM) $(LIBDIR)/libcups.dylib; \
-		$(LN) $(LIBCUPS) $(LIBDIR)/libcups.dylib; \
-	fi
-	if test $(LIBCUPS) = "libcups_s.a"; then \
-		$(INSTALL_LIB) libcups.a $(LIBDIR); \
-	fi
+	$(INSTALL_DIR) $(DESTDIR)$(LIBDIR)
+	$(INSTALL_LIB) $(LIBCUPS) $(DESTDIR)$(LIBDIR)
+	$(LN) $(LIBCUPS) $(DESTDIR)$(LIBDIR)/`basename $(LIBCUPS) .2`
+	$(INSTALL_LIB) libcups.a $(DESTDIR)$(LIBDIR)
 
 
 #
--- cups-1.1.13/data/Makefile.orig	Thu Jan 31 15:53:42 2002
+++ cups-1.1.13/data/Makefile	Tue Feb  5 08:22:43 2002
@@ -85,27 +85,20 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DATADIR)/banners
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/banners
 	for file in $(BANNERS); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/banners; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/banners; \
 	done
-	$(INSTALL_DIR) $(DATADIR)/charsets
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/charsets
 	for file in $(CHARSETS); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/charsets; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/charsets; \
 	done
-	$(INSTALL_DIR) $(DATADIR)/data
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/data
 	for file in $(DATAFILES); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/data; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/data; \
 	done
-	-if test x$(PAMDIR) != x$(BUILDROOT); then \
-		$(INSTALL_DIR) $(PAMDIR); \
-		if test -f /lib/security/pam_unix.so; then \
-			$(INSTALL_DATA) cups.suse $(PAMDIR)/cups; \
-		else \
-			$(INSTALL_DATA) cups.pam $(PAMDIR)/cups; \
-		fi \
-	fi
-
+	$(INSTALL_DIR) $(DESTDIR)$(PAMDIR)
+	$(INSTALL_DATA) cups.pam $(DESTDIR)$(PAMDIR)/cups
 
 #
 # End of "$Id$".
diff -Nur cups-1.1.12/doc/Makefile cups-1.1.12-new/doc/Makefile
--- cups-1.1.12/doc/Makefile	Thu Jul 19 22:17:53 2001
+++ cups-1.1.12-new/doc/Makefile	Sat Nov 24 20:27:51 2001
@@ -118,21 +118,21 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DOCDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(DOCDIR)
 	for file in $(WEBPAGES); do \
-		$(INSTALL_MAN) $$file $(DOCDIR); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR); \
 	done
-	$(INSTALL_MAN) overview.html $(DOCDIR)
-	$(INSTALL_MAN) overview.pdf $(DOCDIR)
+	$(INSTALL_MAN) overview.html $(DESTDIR)$(DOCDIR)
+	$(INSTALL_MAN) overview.pdf $(DESTDIR)$(DOCDIR)
 	for file in $(DOCUMENTS:.shtml=.html); do \
-		$(INSTALL_MAN) $$file $(DOCDIR); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR); \
 	done
 	for file in $(DOCUMENTS:.shtml=.pdf); do \
-		$(INSTALL_MAN) $$file $(DOCDIR); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR); \
 	done
-	$(INSTALL_DIR) $(DOCDIR)/images
+	$(INSTALL_DIR) $(DESTDIR)$(DOCDIR)/images
 	for file in $(WEBIMAGES) $(DOCIMAGES); do \
-		$(INSTALL_MAN) $$file $(DOCDIR)/images; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(DOCDIR)/images; \
 	done
 
 
--- cups-1.1.13/filter/Makefile.orig	Wed Jan 23 23:30:42 2002
+++ cups-1.1.13/filter/Makefile	Tue Feb  5 08:25:37 2002
@@ -59,24 +59,15 @@
 #
 
 install:
-	$(INSTALL_DIR) $(SERVERBIN)/filter
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/filter
 	for file in $(TARGETS); do \
-		$(INSTALL_BIN) $$file $(SERVERBIN)/filter; \
+		$(INSTALL_BIN) $$file $(DESTDIR)$(SERVERBIN)/filter; \
 	done
-	$(INSTALL_DIR) $(LIBDIR)
-	$(INSTALL_LIB) $(LIBCUPSIMAGE) $(LIBDIR)
-	-if test $(LIBCUPSIMAGE) = "libcupsimage.so.2" -o $(LIBCUPSIMAGE) = "libcupsimage.sl.2"; then \
-		$(RM) $(LIBDIR)/`basename $(LIBCUPSIMAGE) .2`; \
-		$(LN) $(LIBCUPSIMAGE) $(LIBDIR)/`basename $(LIBCUPSIMAGE) .2`; \
-	fi
-	-if test $(LIBCUPSIMAGE) = "libcupsimage.2.dylib"; then \
-		$(RM) $(LIBDIR)/libcupsimage.dylib; \
-		$(LN) $(LIBCUPSIMAGE) $(LIBDIR)/libcupsimage.dylib; \
-	fi
-	$(INSTALL_DIR) $(INCLUDEDIR)/cups
-	$(INSTALL_DATA) image.h $(INCLUDEDIR)/cups
-	$(INSTALL_DATA) raster.h $(INCLUDEDIR)/cups
-
+	$(INSTALL_DIR) $(DESTDIR)$(LIBDIR)
+	$(INSTALL_LIB) $(LIBCUPSIMAGE) $(DESTDIR)$(LIBDIR)
+	$(LN) $(LIBCUPSIMAGE) $(DESTDIR)$(LIBDIR)/`basename $(LIBCUPSIMAGE) .2`; \
+	$(INSTALL_DIR) $(DESTDIR)$(INCLUDEDIR)/cups
+	$(INSTALL_DATA) image.h raster.h $(DESTDIR)$(INCLUDEDIR)/cups
 
 #
 # formtops
diff -Nur cups-1.1.12/fonts/Makefile cups-1.1.12-new/fonts/Makefile
--- cups-1.1.12/fonts/Makefile	Thu Jul 19 22:17:55 2001
+++ cups-1.1.12-new/fonts/Makefile	Sat Nov 24 20:27:51 2001
@@ -64,9 +64,9 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DATADIR)/fonts
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/fonts
 	for file in $(FONTS); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/fonts; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/fonts; \
 	done
 
 
diff -Nur cups-1.1.12/locale/Makefile cups-1.1.12-new/locale/Makefile
--- cups-1.1.12/locale/Makefile	Tue Oct 23 19:20:23 2001
+++ cups-1.1.12-new/locale/Makefile	Sat Nov 24 20:27:51 2001
@@ -50,10 +50,10 @@
 #
 
 install:
-	$(INSTALL_DIR) $(LOCALEDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(LOCALEDIR)
 	for dir in $(LOCALES) ; do \
-		$(INSTALL_DIR) $(LOCALEDIR)/$$dir ; \
-		$(INSTALL_DATA) $$dir/cups_$$dir $(LOCALEDIR)/$$dir ; \
+		$(INSTALL_DIR) $(DESTDIR)$(LOCALEDIR)/$$dir ; \
+		$(INSTALL_DATA) $$dir/cups_$$dir $(DESTDIR)$(LOCALEDIR)/$$dir ; \
 	done
 
 
diff -Nur cups-1.1.12/man/Makefile cups-1.1.12-new/man/Makefile
--- cups-1.1.12/man/Makefile	Fri Oct 26 05:16:48 2001
+++ cups-1.1.12-new/man/Makefile	Sat Nov 24 20:48:12 2001
@@ -63,50 +63,25 @@
 #
 
 install:
-	$(INSTALL_DIR) $(MANDIR)/man1
+	$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man1
 	for file in $(MAN1); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/man1/`basename $$file man`1; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(MANDIR)/man1/`basename $$file man`1; \
 	done
-	$(RM) $(MANDIR)/man1/cancel.1
-	$(LN) lp.1 $(MANDIR)/man1/cancel.1
-	$(INSTALL_DIR) $(PMANDIR)/man3
+	$(INSTALL_DIR) $(DESTDIR)$(PMANDIR)/man3
 	for file in $(MAN3); do \
-		$(INSTALL_MAN) $$file $(PMANDIR)/man3/`basename $$file man`3; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(MANDIR)/man3/`basename $$file man`3; \
 	done
-	$(INSTALL_DIR) $(MANDIR)/man5
+	$(INSTALL_DIR) $(DESTDIR)$(PMANDIR)/man5
 	for file in $(MAN5); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/man5/`basename $$file man`5; \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(MANDIR)/man5/`basename $$file man`5; \
 	done
-	$(INSTALL_DIR) $(AMANDIR)/man$(MAN8EXT)
+	$(INSTALL_DIR) $(DESTDIR)$(PMANDIR)/man8
 	for file in $(MAN8); do \
-		$(INSTALL_MAN) $$file $(AMANDIR)/man$(MAN8EXT)/`basename $$file man`$(MAN8EXT); \
+		$(INSTALL_MAN) $$file $(DESTDIR)$(AMANDIR)/man$(MAN8EXT)/`basename $$file man`$(MAN8EXT); \
 	done
-	$(RM) $(AMANDIR)/man$(MAN8EXT)/reject.$(MAN8EXT)
-	$(LN) accept.$(MAN8EXT) $(AMANDIR)/man$(MAN8EXT)/reject.$(MAN8EXT)
-	$(RM) $(AMANDIR)/man$(MAN8EXT)/disable.$(MAN8EXT)
-	$(LN) enable.$(MAN8EXT) $(AMANDIR)/man$(MAN8EXT)/disable.$(MAN8EXT)
-	$(INSTALL_DIR) $(MANDIR)/cat1
-	for file in $(CAT1); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/cat1; \
-	done
-	$(RM) $(MANDIR)/cat1/cancel.$(CAT1EXT)
-	$(LN) lp.$(CAT1EXT) $(MANDIR)/cat1/cancel.$(CAT1EXT)
-	$(INSTALL_DIR) $(PMANDIR)/cat3
-	for file in $(CAT3); do \
-		$(INSTALL_MAN) $$file $(PMANDIR)/cat3; \
-	done
-	$(INSTALL_DIR) $(MANDIR)/cat5
-	for file in $(CAT5); do \
-		$(INSTALL_MAN) $$file $(MANDIR)/cat5; \
-	done
-	$(INSTALL_DIR) $(AMANDIR)/cat$(MAN8EXT)
-	for file in $(CAT8); do \
-		$(INSTALL_MAN) $$file $(AMANDIR)/cat$(MAN8EXT); \
-	done
-	$(RM) $(AMANDIR)/cat$(MAN8EXT)/reject.$(CAT8EXT)
-	$(LN) accept.$(CAT8EXT) $(AMANDIR)/cat$(MAN8EXT)/reject.$(CAT8EXT)
-	$(RM) $(AMANDIR)/cat$(MAN8EXT)/disable.$(CAT8EXT)
-	$(LN) enable.$(CAT8EXT) $(AMANDIR)/cat$(MAN8EXT)/disable.$(CAT8EXT)
+	echo ".so accept.$(MAN8EXT)" > $(DESTDIR)$(AMANDIR)/man$(MAN8EXT)/reject.$(MAN8EXT)
+	echo ".so enable.$(MAN8EXT)" > $(DESTDIR)$(AMANDIR)/man$(MAN8EXT)/disable.$(MAN8EXT)
+
 
 
 #
diff -Nur cups-1.1.12/pdftops/Makefile cups-1.1.12-new/pdftops/Makefile
--- cups-1.1.12/pdftops/Makefile	Wed Jun 27 23:53:16 2001
+++ cups-1.1.12-new/pdftops/Makefile	Sat Nov 24 20:27:51 2001
@@ -43,8 +43,8 @@
 #
 
 install:
-	$(INSTALL_DIR) $(SERVERBIN)/filter
-	$(INSTALL_BIN) pdftops $(SERVERBIN)/filter
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/filter
+	$(INSTALL_BIN) pdftops $(DESTDIR)$(SERVERBIN)/filter
 
 
 #
diff -Nur cups-1.1.12/ppd/Makefile cups-1.1.12-new/ppd/Makefile
--- cups-1.1.12/ppd/Makefile	Thu Jul 19 22:17:56 2001
+++ cups-1.1.12-new/ppd/Makefile	Sat Nov 24 20:27:51 2001
@@ -52,9 +52,9 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DATADIR)/model
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/model
 	for file in $(FILES); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/model; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/model; \
 	done
 
 
diff -Nur cups-1.1.12/pstoraster/Makefile cups-1.1.12-new/pstoraster/Makefile
--- cups-1.1.12/pstoraster/Makefile	Wed Oct 31 17:18:33 2001
+++ cups-1.1.12-new/pstoraster/Makefile	Sat Nov 24 20:27:51 2001
@@ -376,11 +376,11 @@
 #
 
 install:	$(TARGETS)
-	$(INSTALL_DIR) $(SERVERBIN)/filter
-	$(INSTALL_BIN) pstoraster $(SERVERBIN)/filter
-	$(INSTALL_DIR) $(DATADIR)/pstoraster
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/filter
+	$(INSTALL_BIN) pstoraster $(DESTDIR)$(SERVERBIN)/filter
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/pstoraster
 	for file in $(DFILES); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/pstoraster; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/pstoraster; \
 	done
 
 
diff -Nur cups-1.1.12/scheduler/Makefile cups-1.1.12-new/scheduler/Makefile
--- cups-1.1.12/scheduler/Makefile	Mon Jul 23 20:48:52 2001
+++ cups-1.1.12-new/scheduler/Makefile	Sat Nov 24 20:27:51 2001
@@ -52,17 +52,17 @@
 #
 
 install:
-	$(INSTALL_DIR) $(SBINDIR)
-	$(INSTALL_BIN) cupsd $(SBINDIR)
-	$(INSTALL_DIR) $(SERVERBIN)/daemon
-	$(INSTALL_BIN) cups-lpd $(SERVERBIN)/daemon
-	$(INSTALL_BIN) cups-polld $(SERVERBIN)/daemon
-	$(INSTALL_DIR) -m 711 -o $(CUPS_USER) -g $(CUPS_GROUP) $(SERVERROOT)/certs
-	$(INSTALL_DIR) $(SERVERROOT)/interfaces
-	$(INSTALL_DIR) $(SERVERROOT)/ppd
-	$(INSTALL_DIR) $(LOGDIR)
-	$(INSTALL_DIR) -m 700 -o $(CUPS_USER) -g $(CUPS_GROUP) $(REQUESTS)
-	$(INSTALL_DIR) -m 1700 -o $(CUPS_USER) -g $(CUPS_GROUP) $(REQUESTS)/tmp
+	$(INSTALL_DIR) $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) cupsd $(DESTDIR)$(SBINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERBIN)/daemon
+	$(INSTALL_BIN) cups-lpd $(DESTDIR)$(SERVERBIN)/daemon
+	$(INSTALL_BIN) cups-polld $(DESTDIR)$(SERVERBIN)/daemon
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERROOT)/certs
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERROOT)/interfaces
+	$(INSTALL_DIR) $(DESTDIR)$(SERVERROOT)/ppd
+	$(INSTALL_DIR) $(DESTDIR)$(LOGDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(REQUESTS)
+	$(INSTALL_DIR) $(DESTDIR)$(REQUESTS)/tmp
 
 
 #
diff -Nur cups-1.1.12/systemv/Makefile cups-1.1.12-new/systemv/Makefile
--- cups-1.1.12/systemv/Makefile	Tue Sep 18 21:14:47 2001
+++ cups-1.1.12-new/systemv/Makefile	Sat Nov 24 20:39:57 2001
@@ -50,23 +50,20 @@
 #
 
 install:
-	$(INSTALL_DIR) $(BINDIR)
-	$(INSTALL_BIN) accept $(SBINDIR)
-	$(RM) $(SBINDIR)/reject
-	$(LN) accept $(SBINDIR)/reject
-	$(INSTALL_BIN) cupsaddsmb $(SBINDIR)
-	$(INSTALL_BIN) lpadmin $(SBINDIR)
-	$(INSTALL_BIN) lpinfo $(SBINDIR)
-	$(INSTALL_BIN) lpmove $(SBINDIR)
-	$(INSTALL_BIN) cancel $(BINDIR)
-	$(RM) $(BINDIR)/disable
-	$(LN) ../sbin/accept $(BINDIR)/disable
-	$(RM) $(BINDIR)/enable
-	$(LN) ../sbin/accept $(BINDIR)/enable
-	$(INSTALL_BIN) lp $(BINDIR)
-	$(INSTALL_BIN) lpoptions $(BINDIR)
-	$(INSTALL_BIN) lpstat $(BINDIR)
-	$(INSTALL_BIN) -m 4755 -o $(CUPS_USER) -g $(CUPS_GROUP) lppasswd $(BINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) accept $(DESTDIR)$(SBINDIR)
+	$(LN) -sf accept $(DESTDIR)$(SBINDIR)/reject
+	$(INSTALL_BIN) cupsaddsmb $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) lpadmin $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) lpinfo $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) lpmove $(DESTDIR)$(SBINDIR)
+	$(INSTALL_BIN) cancel $(DESTDIR)$(BINDIR)
+	$(LN) -sf ../sbin/accept $(DESTDIR)$(BINDIR)/disable
+	$(LN) -sf ../sbin/accept $(DESTDIR)$(BINDIR)/enable
+	$(INSTALL_BIN) lp $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lpoptions $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lpstat $(DESTDIR)$(BINDIR)
+	$(INSTALL_BIN) lppasswd $(DESTDIR)$(BINDIR)
 
 
 #
diff -Nur cups-1.1.12/templates/Makefile cups-1.1.12-new/templates/Makefile
--- cups-1.1.12/templates/Makefile	Mon Sep 24 19:51:16 2001
+++ cups-1.1.12-new/templates/Makefile	Sat Nov 24 20:27:51 2001
@@ -92,9 +92,9 @@
 #
 
 install:
-	$(INSTALL_DIR) $(DATADIR)/templates
+	$(INSTALL_DIR) $(DESTDIR)$(DATADIR)/templates
 	for file in $(FILES); do \
-		$(INSTALL_DATA) $$file $(DATADIR)/templates; \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(DATADIR)/templates; \
 	done
 
 
