Index: cups/language.c
===================================================================
--- cups/language.c	(revision 10320)
+++ cups/language.c	(revision 10321)
@@ -3,7 +3,7 @@
  *
  *   I18N/language support for CUPS.
  *
- *   Copyright 2007-2011 by Apple Inc.
+ *   Copyright 2007-2012 by Apple Inc.
  *   Copyright 1997-2007 by Easy Software Products.
  *
  *   These coded instructions, statements, and computer programs are the
@@ -979,8 +979,23 @@
       */
 
       if (m)
-        cupsArrayAdd(a, m);
+      {
+        if (m->str[0])
+        {
+          cupsArrayAdd(a, m);
+        }
+        else
+        {
+         /*
+          * Translation is empty, don't add it... (STR #4033)
+          */
 
+          free(m->id);
+          free(m->str);
+          free(m);
+        }
+      }
+
      /*
       * Create a new message with the given msgid string...
       */
@@ -1057,8 +1072,23 @@
   */
 
   if (m)
-    cupsArrayAdd(a, m);
+  {
+    if (m->str[0])
+    {
+      cupsArrayAdd(a, m);
+    }
+    else
+    {
+     /*
+      * Translation is empty, don't add it... (STR #4033)
+      */
 
+      free(m->id);
+      free(m->str);
+      free(m);
+    }
+  }
+
  /*
   * Close the message catalog file and return the new array...
   */
