diff -u -r cups-1.2.1.orig/templates/pl/printer-accept.tmpl cups-1.2.1/templates/pl/printer-accept.tmpl
--- cups-1.2.1.orig/templates/pl/printer-accept.tmpl	2006-05-15 22:06:46.000000000 +0200
+++ cups-1.2.1/templates/pl/printer-accept.tmpl	2006-06-02 20:07:26.000000000 +0200
@@ -1,3 +1,3 @@
 <P>{is_class?Klasa:Drukarka} <A
-HREF="/{is_class?klasy:drukarki}/{printer_name}">{printer_name}</A>
+HREF="/{is_class?classes:printers}/{printer_name}">{printer_name}</A>
 teraz akceptuje zadania.</P>
diff -u -r cups-1.2.1.orig/templates/pl/printer-default.tmpl cups-1.2.1/templates/pl/printer-default.tmpl
--- cups-1.2.1.orig/templates/pl/printer-default.tmpl	2006-05-15 22:06:46.000000000 +0200
+++ cups-1.2.1/templates/pl/printer-default.tmpl	2006-06-02 20:08:36.000000000 +0200
@@ -1,5 +1,5 @@
 <P>{is_class?Klasa:Drukarka} <A
-HREF="/{is_class?klasy:drukarki}/{printer_name}">{printer_name}</A>
+HREF="/{is_class?classes:printers}/{printer_name}">{printer_name}</A>
 została utworzona domyślną drukarką na serwerze.</P>
 
 <BLOCKQUOTE><B>Uwaga:</B> każdy domyślny użytkownik, który został ustawiony przez
diff -u -r cups-1.2.1.orig/templates/pl/printer-purge.tmpl cups-1.2.1/templates/pl/printer-purge.tmpl
--- cups-1.2.1.orig/templates/pl/printer-purge.tmpl	2006-05-15 22:06:46.000000000 +0200
+++ cups-1.2.1/templates/pl/printer-purge.tmpl	2006-06-02 20:09:23.000000000 +0200
@@ -1,3 +1,3 @@
 <P>{is_class?Klasa:Drukarka} <A
-HREF="/{is_class?klasy:drukarki/{printer_name}">{printer_name}</A>
+HREF="/{is_class?classes:printers/{printer_name}">{printer_name}</A>
 została wyczyszczone ze wszystkich zadań.</P>
diff -u -r cups-1.2.1.orig/templates/pl/printer-reject.tmpl cups-1.2.1/templates/pl/printer-reject.tmpl
--- cups-1.2.1.orig/templates/pl/printer-reject.tmpl	2006-05-15 22:06:46.000000000 +0200
+++ cups-1.2.1/templates/pl/printer-reject.tmpl	2006-06-02 20:10:41.000000000 +0200
@@ -1,3 +1,3 @@
 <P>{is_class?Klasa:Drukarka} <A
-HREF="/{is_class?klasy:drukarki}/{printer_name}">{printer_name}</A>
+HREF="/{is_class?classes:printers}/{printer_name}">{printer_name}</A>
 już nie akceptuje zadań.</P>
diff -u -r cups-1.2.1.orig/templates/pl/printer-start.tmpl cups-1.2.1/templates/pl/printer-start.tmpl
--- cups-1.2.1.orig/templates/pl/printer-start.tmpl	2006-05-15 22:06:46.000000000 +0200
+++ cups-1.2.1/templates/pl/printer-start.tmpl	2006-06-02 20:11:12.000000000 +0200
@@ -1,3 +1,3 @@
 <P>{is_class?Klasa:Drukarka} <A
-HREF="/{is_class?klasy:drukarki}/{printer_name}">{printer_name}</A>
+HREF="/{is_class?classes:printers}/{printer_name}">{printer_name}</A>
 została uruchomiona.</P>
diff -u -r cups-1.2.1.orig/templates/pl/printer-stop.tmpl cups-1.2.1/templates/pl/printer-stop.tmpl
--- cups-1.2.1.orig/templates/pl/printer-stop.tmpl	2006-05-15 22:06:46.000000000 +0200
+++ cups-1.2.1/templates/pl/printer-stop.tmpl	2006-06-02 20:20:12.000000000 +0200
@@ -1,3 +1,3 @@
 <P>{is_class?Klasa:Drukarka} <A
-HREF="/{is_class?klasy:drukarki}/{printer_name}">{printer_name}</A>
+HREF="/{is_class?classes:printers}/{printer_name}">{printer_name}</A>
 została zatrzymana.</P>
--- cups-1.2.8/templates/pl/printers.tmpl.orig	2006-05-15 22:06:46.000000000 +0200
+++ cups-1.2.8/templates/pl/printers.tmpl	2007-02-18 10:12:35.967716799 +0100
@@ -6,7 +6,7 @@
 <TABLE WIDTH="100%" CLASS="button" CELLSPACING="0" CELLPADDING="0" SUMMARY="{printer_name}">
 <TR>
 <TD VALIGN="TOP"><A HREF="{printer_uri_supported}">
-<IMG SRC="/images/printer-{printer_state=3?bezczynna:{printer_state=4?przetwarzanie:zatrzymana}}.gif" CLASS="button" ALT=""></A>
+<IMG SRC="/images/printer-{printer_state=3?idle:{printer_state=4?processing:stopped}}.gif" CLASS="button" ALT=""></A>
 </TD>
 <TD VALIGN="TOP"><B>Opis:</B> {printer_info}<BR>
 <B>Położenie:</B> {printer_location}<BR>
