--- cups-1.1.23/scheduler/ipp.c.orig	2005-01-03 20:29:59.000000000 +0100
+++ cups-1.1.23/scheduler/ipp.c	2005-10-17 14:20:24.000000000 +0200
@@ -6792,7 +6792,9 @@
   struct passwd		*user;		/* User info */
   struct group		*group;		/* System group info */
   char			junk[33];	/* MD5 password (not used) */
+  location_t		*loc;		/* Authentication location */
 
+  loc = FindBest(con->uri, con->http.state);
 
   LogMessage(L_DEBUG2, "validate_user(%p[%d], \"%s\", %p, %d)\n",
              con, con->http.fd, owner, username, userlen);
@@ -6801,6 +6803,9 @@
   * Validate input...
   */
 
+  if (loc->type == AUTH_NONE)
+    return (1);
+
   if (con == NULL || owner == NULL || username == NULL || userlen <= 0)
     return (0);
 
