--- cups-2.3.6/backend/ipp.c.orig	2024-05-19 14:12:43.566004590 +0200
+++ cups-2.3.6/backend/ipp.c	2024-05-19 14:49:20.758512651 +0200
@@ -1519,14 +1519,13 @@
 
   if (create_job)
   {
-    monitor.job_name = argv[3];
+    snprintf(print_job_name, 127, "%s", argv[3]);
   }
   else
   {
-    snprintf(print_job_name, sizeof(print_job_name), "%s - %s", argv[1],
-             argv[3]);
-    monitor.job_name = print_job_name;
+    snprintf(print_job_name, 127, "%s - %s", argv[1], argv[3]);
   }
+  monitor.job_name = print_job_name;
 
   _cupsThreadCreate((_cups_thread_func_t)monitor_printer, &monitor);
 
